// @elastic/eui/es/services/hooks/useIsPointerDown.js
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { useState, useEffect } from 'react';

/**
 * A hook that tracks whether the pointer is currently down/pressed.
 * Useful for detecting text selection in progress.
 */
export function useIsPointerDown(container) {
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isPointerDown = _useState2[0],
    setIsPointerDown = _useState2[1];
  useEffect(function () {
    var handlePointerDown = function handlePointerDown(event) {
      if (container !== null && container !== void 0 && container.current && !container.current.contains(event.target)) {
        return;
      }
      setIsPointerDown(true);
    };
    var handlePointerUpOrCancel = function handlePointerUpOrCancel() {
      setIsPointerDown(false);
    };
    var handleVisibilityChange = function handleVisibilityChange() {
      if (document.visibilityState === 'hidden') {
        setIsPointerDown(false);
      }
    };
    var controller = new AbortController();
    var options = {
      capture: true,
      signal: controller.signal
    };
    document.addEventListener('pointerdown', handlePointerDown, options);
    document.addEventListener('pointerup', handlePointerUpOrCancel, options);
    document.addEventListener('pointercancel', handlePointerUpOrCancel, options);
    document.addEventListener('visibilitychange', handleVisibilityChange, {
      signal: controller.signal
    });
    return function () {
      controller.abort();
    };
  }, [container]);
  return isPointerDown;
}
