// @elastic/eui/optimize/es/components/datagrid/body/header/draggable_columns.styles.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
var _templateObject;
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css, keyframes } from '@emotion/react';
import { euiCanAnimate, logicalCSS } from '../../../../global_styling';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "pfqu74-euiDataGridHeaderDroppable",
  styles: "display:flex;label:euiDataGridHeaderDroppable;"
} : {
  name: "pfqu74-euiDataGridHeaderDroppable",
  styles: "display:flex;label:euiDataGridHeaderDroppable;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiDataGridDraggableHeaderStyles = function euiDataGridDraggableHeaderStyles(_ref2) {
  var euiTheme = _ref2.euiTheme;
  return {
    euiDataGridHeaderDroppable: _ref,
    // The resizer must be positioned outside the draggable component to ensure both work independently
    euiDataGridHeaderCellDraggableWrapper: /*#__PURE__*/css("position:relative;.euiDataGridColumnResizer::after{", logicalCSS('margin-left', "-".concat(euiTheme.border.width.thick)), ";};label:euiDataGridHeaderCellDraggableWrapper;"),
    // override internal styling from @hello-pangea/dnd to ensure positioning
    euiDataGridHeaderCellDraggable: /*#__PURE__*/css("display:flex;", logicalCSS('height', '100%'), ";;label:euiDataGridHeaderCellDraggable;"),
    // Add more visual affordance to keyboard drags (raises cell slightly to show green droppable bg)
    // Using animation as transition doesn't seem to work (a tale as old as EuiDataGrid...)
    isKeyboardDragging: /*#__PURE__*/css("animation-name:", keyframes(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n        from { transform: translateY(0); }\n        to { transform: translateY(-", "); }\n      "])), euiTheme.size.s), ";animation-iteration-count:1;animation-fill-mode:forwards;", euiCanAnimate, "{animation-duration:", euiTheme.animation.fast, ";};label:isKeyboardDragging;"),
    // Ensure correct cell background colors on drag
    underline: /*#__PURE__*/css("background-color:", euiTheme.colors.emptyShade, ";;label:underline;"),
    shade: /*#__PURE__*/css("background-color:", euiTheme.colors.lightestShade, ";;label:shade;"),
    noLeadingBorder: /*#__PURE__*/css(logicalCSS('border-left', 'none !important'), ";;label:noLeadingBorder;")
  };
};
