// @elastic/eui/optimize/lib/components/list_group/pinnable_list_group/pinnable_list_group.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EuiPinnableListGroup = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _services = require("../../../services");
var _i18n = require("../../i18n");
var _list_group = require("../list_group");
var _pinnable_list_group = require("./pinnable_list_group.styles");
var _react2 = require("@emotion/react");
var _excluded = ["className", "listItems", "pinTitle", "unpinTitle", "onPinClick"],
  _excluded2 = ["pinned", "pinnable"];
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var EuiPinnableListGroup = exports.EuiPinnableListGroup = function EuiPinnableListGroup(_ref) {
  var className = _ref.className,
    listItems = _ref.listItems,
    pinTitle = _ref.pinTitle,
    unpinTitle = _ref.unpinTitle,
    onPinClick = _ref.onPinClick,
    rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var classes = (0, _classnames.default)('euiPinnableListGroup', className);
  var styles = (0, _services.useEuiMemoizedStyles)(_pinnable_list_group.euiPinnableListGroupItemExtraActionStyles);
  var pinExtraActionLabel = (0, _i18n.useEuiI18n)('euiPinnableListGroup.pinExtraActionLabel', 'Pin item');
  var pinnedExtraActionLabel = (0, _i18n.useEuiI18n)('euiPinnableListGroup.pinnedExtraActionLabel', 'Unpin item');

  // Alter listItems object with extra props
  var pinnableListItems = (0, _react.useMemo)(function () {
    return listItems.map(function (item) {
      var pinned = item.pinned,
        _item$pinnable = item.pinnable,
        pinnable = _item$pinnable === void 0 ? true : _item$pinnable,
        itemProps = (0, _objectWithoutProperties2.default)(item, _excluded2);
      // Make some declarations of props for the nav implementation
      itemProps.className = (0, _classnames.default)('euiPinnableListGroup__item', item.className);

      // Add the pinning action unless the item has its own extra action
      if (pinnable && !itemProps.extraAction) {
        // Different displays for pinned vs unpinned
        var sharedProps = {
          iconType: 'pinFilled',
          iconSize: 's',
          css: [styles.euiPinnableListGroup__itemExtraAction, pinned && styles.pinned]
        };
        if (pinned) {
          var title = unpinTitle ? unpinTitle(item) : pinnedExtraActionLabel;
          itemProps.extraAction = _objectSpread(_objectSpread({}, sharedProps), {}, {
            alwaysShow: true,
            'aria-label': title,
            title: title
          });
        } else {
          var _title = pinTitle ? pinTitle(item) : pinExtraActionLabel;
          itemProps.extraAction = _objectSpread(_objectSpread({}, sharedProps), {}, {
            'aria-label': _title,
            title: _title
          });
        }
        // Return the item on click
        itemProps.extraAction.onClick = function () {
          return onPinClick(item);
        };
      }
      return itemProps;
    });
  }, [listItems, pinTitle, pinExtraActionLabel, unpinTitle, pinnedExtraActionLabel, onPinClick, styles]);
  return (0, _react2.jsx)(_list_group.EuiListGroup, (0, _extends2.default)({
    className: classes,
    listItems: pinnableListItems
  }, rest));
};
