// @elastic/eui/lib/components/button/split_button/split_button.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._EuiSplitButton = exports.EuiSplitButton = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _services = require("../../../services");
var _global_styling = require("../../../global_styling");
var _split_button_context = require("./split_button_context");
var _split_button_actions = require("./split_button_actions");
var _split_button = require("./split_button.styles");
var _react2 = require("@emotion/react");
var _excluded = ["className", "children", "size", "color", "fill", "isDisabled", "hasAriaDisabled", "isLoading"];
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], t.indexOf(o) >= 0 || {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.indexOf(n) >= 0) continue; t[n] = r[n]; } return t; }
var _EuiSplitButton = exports._EuiSplitButton = function _EuiSplitButton(_ref) {
  var className = _ref.className,
    children = _ref.children,
    _ref$size = _ref.size,
    size = _ref$size === void 0 ? 'm' : _ref$size,
    _ref$color = _ref.color,
    color = _ref$color === void 0 ? 'primary' : _ref$color,
    _ref$fill = _ref.fill,
    fill = _ref$fill === void 0 ? false : _ref$fill,
    isDisabled = _ref.isDisabled,
    hasAriaDisabled = _ref.hasAriaDisabled,
    isLoading = _ref.isLoading,
    rest = _objectWithoutProperties(_ref, _excluded);
  var euiThemeContext = (0, _services.useEuiTheme)();
  var highContrastMode = euiThemeContext.highContrastMode;
  var _children = _slicedToArray(children, 2),
    primaryAction = _children[0],
    secondaryAction = _children[1];
  var key = (0, _services.useGeneratedHtmlId)({
    suffix: 'EuiSplitButton'
  });
  var commonProps = {
    size: size,
    color: color,
    fill: fill,
    isDisabled: isDisabled,
    hasAriaDisabled: hasAriaDisabled,
    isLoading: isLoading
  };
  var buttonFilledColors = (0, _global_styling.getEuiFilledButtonColors)(euiThemeContext, isDisabled ? 'disabled' : color);
  var buttonColors = (0, _global_styling.getEuiButtonColors)(euiThemeContext, isDisabled ? 'disabled' : color);
  var classes = (0, _classnames.default)('euiSplitButton', className);
  var styles = (0, _services.useEuiMemoizedStyles)(_split_button.euiSplitButtonStyles);
  var cssStyles = [styles.euiSplitButton, fill && styles.fill];
  var dividerStyles = (0, _react.useMemo)(function () {
    return (0, _split_button.euiSplitButtonDividerStyles)(euiThemeContext, !fill ? buttonColors.borderColor : highContrastMode && fill ? buttonFilledColors.backgroundColor : 'transparent');
  }, [euiThemeContext, highContrastMode, fill, buttonFilledColors.backgroundColor, buttonColors.borderColor]);

  // NOTE: dev-mode-only runtime check to evaluate if correct child components are passed
  if (process.env.NODE_ENV !== 'production') {
    var childArray = _react.Children.toArray(children);
    var expectedTypes = [['EuiSplitButton.ActionPrimary', 'EuiSplitButtonActionPrimary'], ['EuiSplitButton.ActionSecondary', 'EuiSplitButtonActionSecondary']];
    childArray.forEach(function (child, index) {
      if (! /*#__PURE__*/(0, _react.isValidElement)(child)) return;
      var componentName = getComponentName(child.type);
      var expectedComponents = expectedTypes[index];
      if (!expectedComponents.includes(componentName)) {
        console.warn("\u26A0\uFE0F EuiSplitButton: Expected <".concat(expectedComponents[0], "> at position ").concat(index + 1, ", got <").concat(componentName, ">. You might be using a wrapper. Using e.g. React.memo() or React.lazy() is valid, other component wrappers are not and will break styling.\n          To verify expected usage, please check the documentation: https://eui.elastic.co/docs/components/navigation/buttons/split-button/"));
      }
    });
  }
  return (0, _react2.jsx)("div", _extends({
    role: "group",
    className: classes,
    css: cssStyles
  }, rest), (0, _react2.jsx)(_split_button_context.EuiSplitButtonContext.Provider, {
    value: commonProps
  }, (0, _react2.jsx)(_react.Fragment, {
    key: "".concat(key, "-primaryAction")
  }, primaryAction), (0, _react2.jsx)("div", {
    className: "euiSplitButton__divider",
    css: dividerStyles.divider,
    "aria-hidden": "true"
  }), (0, _react2.jsx)(_react.Fragment, {
    key: "".concat(key, "-secondaryAction")
  }, secondaryAction)));
};
_EuiSplitButton.propTypes = {
  className: _propTypes.default.string,
  "aria-label": _propTypes.default.string,
  "data-test-subj": _propTypes.default.string,
  css: _propTypes.default.any,
  size: _propTypes.default.any,
  color: _propTypes.default.any,
  fill: _propTypes.default.bool,
  isLoading: _propTypes.default.any,
  /* NOTE: This definition is not actually enforced by Typescript.
      The tuple type ensures 2 children are expected, but the function component type won't be evaluated properly.
      We use this definition anyway as documentation for users.
      To advocate correct usage this requires runtime checks for development mode. */
  children: _propTypes.default.any.isRequired
};
var EuiSplitButton = exports.EuiSplitButton = Object.assign(_EuiSplitButton, {
  ActionPrimary: _split_button_actions.EuiSplitButtonActionPrimary,
  ActionSecondary: _split_button_actions.EuiSplitButtonActionSecondary
});

/* internal utils */

var getComponentName = function getComponentName(type) {
  if (typeof type === 'string') return type;
  return type.displayName || type.name || 'Unknown';
};
