// @elastic/eui/optimize/es/components/flyout/manager/flyout_main.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["id", "pushMinBreakpoint", "type", "side"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { forwardRef } from 'react';
import { EuiManagedFlyout } from './flyout_managed';
import { useHasChildFlyout, useFlyoutId } from './hooks';
import { euiMainFlyoutStyles } from './flyout_main.styles';
import { useEuiMemoizedStyles } from '../../../services';
import { DEFAULT_PUSH_MIN_BREAKPOINT, DEFAULT_SIDE, DEFAULT_TYPE } from '../const';
import { useIsPushed } from '../hooks';
import { LEVEL_MAIN } from './const';

/**
 * Props for `EuiFlyoutMain`, the primary managed flyout component.
 * The `level` prop is fixed internally to `main` and is therefore omitted.
 */
import { jsx as ___EmotionJSX } from "@emotion/react";
/**
 * Managed main flyout. Handles ID management, child-flyout styling,
 * and push/overlay behavior based on provided props.
 */
export var EuiFlyoutMain = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var id = _ref.id,
    _ref$pushMinBreakpoin = _ref.pushMinBreakpoint,
    pushMinBreakpoint = _ref$pushMinBreakpoin === void 0 ? DEFAULT_PUSH_MIN_BREAKPOINT : _ref$pushMinBreakpoin,
    _ref$type = _ref.type,
    type = _ref$type === void 0 ? DEFAULT_TYPE : _ref$type,
    _ref$side = _ref.side,
    side = _ref$side === void 0 ? DEFAULT_SIDE : _ref$side,
    props = _objectWithoutProperties(_ref, _excluded);
  var flyoutId = useFlyoutId(id);
  var hasChildFlyout = useHasChildFlyout(flyoutId);
  var styles = useEuiMemoizedStyles(euiMainFlyoutStyles);
  var isPushed = useIsPushed({
    type: type,
    pushMinBreakpoint: pushMinBreakpoint
  });
  var cssStyles = [hasChildFlyout && !isPushed && styles.hasChildFlyout[side]];
  var style = {};
  return ___EmotionJSX(EuiManagedFlyout, _extends({
    ref: ref,
    id: flyoutId,
    level: LEVEL_MAIN,
    style: style,
    css: cssStyles
  }, _objectSpread(_objectSpread({}, props), {}, {
    pushMinBreakpoint: pushMinBreakpoint,
    type: type,
    side: side
  })));
});
EuiFlyoutMain.displayName = 'EuiFlyoutMain';
