// @elastic/eui/optimize/es/components/flyout/manager/flyout_managed.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["id", "onClose", "onActive", "level", "size", "css", "flyoutMenuProps"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useEffect, useLayoutEffect, useMemo, useRef, useState, forwardRef } from 'react';
import { flushSync } from 'react-dom';
import { useCombinedRefs, useEuiMemoizedStyles } from '../../../services';
import { useEuiI18n } from '../../i18n';
import { useResizeObserver } from '../../observer/resize_observer';
import { EuiFlyoutComponent } from '../flyout.component';
import { EuiFlyoutMenuContext } from '../flyout_menu_context';
import { useFlyoutActivityStage } from './activity_stage';
import { LEVEL_CHILD, LEVEL_MAIN, PROPERTY_FLYOUT, PROPERTY_LAYOUT_MODE, PROPERTY_LEVEL } from './const';
import { EuiFlyoutIsManagedProvider } from './context';
import { euiManagedFlyoutStyles } from './flyout_managed.styles';
import { useFlyoutManager as _useFlyoutManager, useCurrentSession, useFlyoutId, useFlyoutLayoutMode, useIsFlyoutActive, useParentFlyoutSize } from './hooks';
import { useCurrentMainFlyout, useIsFlyoutRegistered } from './selectors';
import { createValidationErrorMessage, isNamedSize, validateManagedFlyoutSize, validateSizeCombination } from './validation';

/**
 * Props for `EuiManagedFlyout`, the internal persistent flyout used by
 * the manager. Extends base flyout props and requires a `level` to
 * distinguish `main` vs `child` behavior.
 */
import { jsx as ___EmotionJSX } from "@emotion/react";
var useFlyoutManager = function useFlyoutManager() {
  var context = _useFlyoutManager();
  if (!context) {
    throw new Error('EuiManagedFlyout must be used within an EuiFlyoutManager');
  }
  return context;
};

/**
 * Persistent managed flyout rendered inside the provider. Handles:
 * - registration/unregistration with the manager
 * - size validation and parent/child size compatibility
 * - width tracking for responsive layouts
 * - lifecycle stage transitions and data attributes for styling
 */
export var EuiManagedFlyout = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var id = _ref.id,
    onCloseProp = _ref.onClose,
    onActiveProp = _ref.onActive,
    level = _ref.level,
    sizeProp = _ref.size,
    customCss = _ref.css,
    _flyoutMenuProps = _ref.flyoutMenuProps,
    props = _objectWithoutProperties(_ref, _excluded);
  var flyoutId = useFlyoutId(id);
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    flyoutRef = _useState2[0],
    setFlyoutRef = _useState2[1];
  var refs = useMemo(function () {
    return [setFlyoutRef, ref];
  }, [ref]);
  var combinedRef = useCombinedRefs(refs);
  var _useFlyoutManager2 = useFlyoutManager(),
    addFlyout = _useFlyoutManager2.addFlyout,
    closeFlyout = _useFlyoutManager2.closeFlyout,
    setFlyoutWidth = _useFlyoutManager2.setFlyoutWidth,
    goBack = _useFlyoutManager2.goBack,
    _historyItems = _useFlyoutManager2.historyItems;
  var parentSize = useParentFlyoutSize(flyoutId);
  var parentFlyout = useCurrentMainFlyout();
  var layoutMode = useFlyoutLayoutMode();
  var styles = useEuiMemoizedStyles(euiManagedFlyoutStyles);

  // Set default size based on level: main defaults to 'm', child defaults to 's'
  var size = sizeProp !== null && sizeProp !== void 0 ? sizeProp : level === LEVEL_CHILD ? 's' : 'm';

  // Validate size
  var sizeTypeError = validateManagedFlyoutSize(size, flyoutId, level);
  if (sizeTypeError) {
    throw new Error(createValidationErrorMessage(sizeTypeError));
  }

  // For child flyouts, validate parent-child combinations
  if (level === LEVEL_CHILD && parentSize && isNamedSize(size) && isNamedSize(parentSize)) {
    var combinationError = validateSizeCombination(parentSize, size);
    if (combinationError) {
      combinationError.flyoutId = flyoutId;
      combinationError.parentFlyoutId = parentFlyout === null || parentFlyout === void 0 ? void 0 : parentFlyout.flyoutId;
      combinationError.level = level;
      throw new Error(createValidationErrorMessage(combinationError));
    }
  }
  var defaultTitle = useEuiI18n('euiFlyoutManaged.defaultTitle', 'Unknown Flyout');

  // Set title from flyoutMenuProps or aria-label
  // TODO: allow aria-labelledby references to be used
  var title = (_flyoutMenuProps === null || _flyoutMenuProps === void 0 ? void 0 : _flyoutMenuProps.title) || props['aria-label'];
  if (process.env.NODE_ENV === 'development' && level === LEVEL_MAIN && !title) {
    console.warn("Managed flyout \"".concat(flyoutId, "\" requires a title, which can be provided through 'flyoutMenuProps.title' or 'aria-label'. Using default title: \"").concat(defaultTitle, "\""));
    title = defaultTitle;
  }
  var isActive = useIsFlyoutActive(flyoutId);
  var currentSession = useCurrentSession();
  var flyoutExistsInManager = useIsFlyoutRegistered(flyoutId);

  // Stabilize the onClose callback
  var onCloseCallbackRef = useRef();
  onCloseCallbackRef.current = function (e) {
    if (onCloseProp) {
      var event = e || new MouseEvent('click');
      onCloseProp(event);
    }
  };

  // Stabilize the onActive callback
  var onActiveCallbackRef = useRef();
  onActiveCallbackRef.current = onActiveProp;

  // Track if flyout was ever registered to avoid false positives on initial mount
  var wasRegisteredRef = useRef(false);

  // Track flyoutExistsInManager in a ref to avoid dependency loop
  // The cleanup function needs the current value but shouldn't cause re-runs
  var flyoutExistsInManagerRef = useRef(flyoutExistsInManager);

  // Update ref when flyoutExistsInManager changes
  useEffect(function () {
    flyoutExistsInManagerRef.current = flyoutExistsInManager;
  }, [flyoutExistsInManager]);

  // Register with flyout manager context when open, remove when closed
  // Using useLayoutEffect to run synchronously before DOM updates
  useLayoutEffect(function () {
    addFlyout(flyoutId, title, level, size);
    return function () {
      // Only call closeFlyout if it wasn't already called via onClose
      // This prevents duplicate removal when using Escape/X button
      if (flyoutExistsInManagerRef.current) {
        closeFlyout(flyoutId);
      }

      // Reset navigation tracking when explicitly closed via isOpen=false
      wasRegisteredRef.current = false;
    };
  }, [flyoutId, title, level, size, addFlyout, closeFlyout]);

  // Detect when flyout has been removed from manager state (e.g., via Back button)
  // and trigger onClose callback to notify the parent component
  useEffect(function () {
    if (flyoutExistsInManager) {
      wasRegisteredRef.current = true;
    }

    // If flyout was previously registered, is marked as open, but no longer exists in manager state,
    // it was removed via navigation (Back button) - trigger close callback
    if (wasRegisteredRef.current && !flyoutExistsInManager) {
      var _onCloseCallbackRef$c;
      (_onCloseCallbackRef$c = onCloseCallbackRef.current) === null || _onCloseCallbackRef$c === void 0 || _onCloseCallbackRef$c.call(onCloseCallbackRef, new MouseEvent('navigation'));
      wasRegisteredRef.current = false; // Reset to avoid repeated calls
    }
  }, [flyoutExistsInManager, flyoutId]);

  // Monitor current session changes and fire onActive callback when this flyout becomes active
  useEffect(function () {
    if (!onActiveCallbackRef.current || !currentSession) {
      return;
    }

    // Make sure callback is only fired for the flyout that changed
    var mainChanged = level === LEVEL_MAIN && currentSession.mainFlyoutId === flyoutId;
    var childChanged = level === LEVEL_CHILD && currentSession.childFlyoutId === flyoutId;
    if (mainChanged || childChanged) {
      onActiveCallbackRef.current();
    }
  }, [currentSession, flyoutId, level]);

  // Track width changes for flyouts
  var _useResizeObserver = useResizeObserver(isActive ? flyoutRef : null, 'width'),
    width = _useResizeObserver.width;

  // Pass the stabilized onClose callback to the flyout menu context
  var onClose = function onClose(e) {
    // CRITICAL: Update manager state FIRST before allowing React to unmount
    // This prevents race conditions during portal → inline DOM transitions
    // and ensures cascade close logic runs before DOM cleanup begins
    // Using flushSync to force synchronous state update completion
    flushSync(function () {
      closeFlyout(flyoutId);
    });

    // trigger parent callback, unmounts the component
    if (onCloseCallbackRef.current) {
      var event = e || new MouseEvent('click');
      onCloseCallbackRef.current(event);
    }
  };

  // Update width in manager state when it changes
  useEffect(function () {
    if (isActive && width) {
      setFlyoutWidth(flyoutId, width);
    }
  }, [flyoutId, level, isActive, width, setFlyoutWidth]);
  var _useFlyoutActivitySta = useFlyoutActivityStage({
      flyoutId: flyoutId,
      level: level
    }),
    activityStage = _useFlyoutActivitySta.activityStage,
    onAnimationEnd = _useFlyoutActivitySta.onAnimationEnd;

  // Note: history controls are only relevant for main flyouts
  var historyItems = useMemo(function () {
    var result = level === LEVEL_MAIN ? _historyItems : undefined;
    return result;
  }, [level, _historyItems]);
  var backButtonProps = useMemo(function () {
    return level === LEVEL_MAIN ? {
      onClick: goBack
    } : undefined;
  }, [level, goBack]);
  var showBackButton = historyItems ? historyItems.length > 0 : false;
  var flyoutMenuProps = _objectSpread(_objectSpread({}, _flyoutMenuProps), {}, {
    historyItems: historyItems,
    showBackButton: showBackButton,
    backButtonProps: backButtonProps,
    title: title
  });
  return ___EmotionJSX(EuiFlyoutIsManagedProvider, {
    isManaged: true
  }, ___EmotionJSX(EuiFlyoutMenuContext.Provider, {
    value: {
      onClose: onClose
    }
  }, ___EmotionJSX(EuiFlyoutComponent, _extends({
    id: flyoutId,
    ref: combinedRef,
    css: [styles.managedFlyout, customCss, styles.stage(activityStage, props.side, level), ";label:EuiManagedFlyout;"]
  }, _objectSpread(_objectSpread({}, props), {}, _defineProperty(_defineProperty(_defineProperty({
    onClose: onClose,
    size: size,
    flyoutMenuProps: flyoutMenuProps,
    onAnimationEnd: onAnimationEnd
  }, PROPERTY_FLYOUT, true), PROPERTY_LAYOUT_MODE, layoutMode), PROPERTY_LEVEL, level))))));
});
EuiManagedFlyout.displayName = 'EuiManagedFlyout';
