// @elastic/eui/es/components/button/split_button/split_button_actions.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["className", "isIconOnly", "tooltipProps"],
  _excluded2 = ["fill", "isDisabled", "isLoading"],
  _excluded3 = ["className", "popoverProps", "tooltipProps"],
  _excluded4 = ["fill", "isDisabled", "isLoading"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], t.indexOf(o) >= 0 || {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.indexOf(n) >= 0) continue; t[n] = r[n]; } return t; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useContext } from 'react';
import PropTypes from "prop-types";
import classNames from 'classnames';
import { EuiPopover } from '../../popover';
import { EuiToolTip } from '../../tool_tip';
import { EuiButton } from '../button';
import { isButtonDisabled } from '../button_display/_button_display';
import { EuiButtonIcon } from '../button_icon';
import { euiSplitButtonActionStyles } from './split_button.styles';
import { EuiSplitButtonContext } from './split_button_context';
import { jsx as ___EmotionJSX } from "@emotion/react";
export var EuiSplitButtonActionPrimary = function EuiSplitButtonActionPrimary(_ref) {
  var className = _ref.className,
    isIconOnly = _ref.isIconOnly,
    tooltipProps = _ref.tooltipProps,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useContext = useContext(EuiSplitButtonContext),
    fill = _useContext.fill,
    isDisabled = _useContext.isDisabled,
    isLoading = _useContext.isLoading,
    sharedRest = _objectWithoutProperties(_useContext, _excluded2);
  var _isDisabled = !!isDisabled || isButtonDisabled(rest);
  var _isLoading = !!isLoading || !!rest.isLoading;
  var display = fill ? 'fill' : 'base';
  var classes = classNames('euiSplitButtonActionPrimary', className);
  var styles = euiSplitButtonActionStyles;
  var actionProps = _objectSpread(_objectSpread(_objectSpread({}, rest), sharedRest), {}, {
    isDisabled: _isDisabled,
    isLoading: _isLoading,
    css: [styles.euiSplitButtonActionPrimary],
    className: classes
  });
  var actionButtonProps = _objectSpread(_objectSpread({}, actionProps), {}, {
    fill: fill
  });
  var actionIconProps = _objectSpread(_objectSpread({}, actionProps), {}, {
    display: display
  });
  var button = isIconOnly ? ___EmotionJSX(EuiButtonIcon, actionIconProps) : ___EmotionJSX(EuiButton, actionButtonProps);
  return tooltipProps ? ___EmotionJSX(EuiToolTip, tooltipProps, button) : button;
};
EuiSplitButtonActionPrimary.propTypes = {
  /**
     * Enables rendering an `EuiToolTip` with the passed props.
     */
  tooltipProps: PropTypes.any,
  /**
       * Toggles the render as `EuiButtonIcon`.
       */
  isIconOnly: PropTypes.bool
};
export var EuiSplitButtonActionSecondary = function EuiSplitButtonActionSecondary(_ref2) {
  var className = _ref2.className,
    popoverProps = _ref2.popoverProps,
    tooltipProps = _ref2.tooltipProps,
    rest = _objectWithoutProperties(_ref2, _excluded3);
  var _useContext2 = useContext(EuiSplitButtonContext),
    fill = _useContext2.fill,
    isDisabled = _useContext2.isDisabled,
    isLoading = _useContext2.isLoading,
    sharedRest = _objectWithoutProperties(_useContext2, _excluded4);
  var _isDisabled = !!isDisabled || isButtonDisabled(rest);
  var _isLoading = !!isLoading || !!rest.isLoading;
  var display = fill ? 'fill' : 'base';
  var classes = classNames('euiSplitButtonActionSecondary', className);
  var styles = euiSplitButtonActionStyles;
  var actionProps = _objectSpread(_objectSpread(_objectSpread({}, rest), sharedRest), {}, {
    display: display,
    // enforce arrowDown icon when a popover is rendered
    iconType: popoverProps != null ? 'arrowDown' : rest.iconType,
    isDisabled: _isDisabled,
    isLoading: _isLoading,
    css: [styles.euiSplitButtonActionSecondary],
    className: classes
  });
  var button = ___EmotionJSX(EuiButtonIcon, actionProps);
  var action = tooltipProps ? ___EmotionJSX(EuiToolTip, tooltipProps, button) : button;
  return popoverProps ? ___EmotionJSX(EuiPopover, _extends({
    anchorPosition: "downCenter"
  }, popoverProps, {
    button: action
  })) : action;
};
EuiSplitButtonActionSecondary.propTypes = {
  /**
     * Enables rendering an `EuiToolTip` with the passed props.
     */
  tooltipProps: PropTypes.any,
  /**
       * Enables rendering an `EuiPopover` with the passed props.
       * When passed the secondary action icon will be fixed to `arrowDown`.
       */
  popoverProps: PropTypes.any
};
