// @elastic/eui/optimize/es/components/list_group/pinnable_list_group/pinnable_list_group.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["className", "listItems", "pinTitle", "unpinTitle", "onPinClick"],
  _excluded2 = ["pinned", "pinnable"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useMemo } from 'react';
import classNames from 'classnames';
import { useEuiMemoizedStyles } from '../../../services';
import { useEuiI18n } from '../../i18n';
import { EuiListGroup } from '../list_group';
import { euiPinnableListGroupItemExtraActionStyles } from './pinnable_list_group.styles';
import { jsx as ___EmotionJSX } from "@emotion/react";
export var EuiPinnableListGroup = function EuiPinnableListGroup(_ref) {
  var className = _ref.className,
    listItems = _ref.listItems,
    pinTitle = _ref.pinTitle,
    unpinTitle = _ref.unpinTitle,
    onPinClick = _ref.onPinClick,
    rest = _objectWithoutProperties(_ref, _excluded);
  var classes = classNames('euiPinnableListGroup', className);
  var styles = useEuiMemoizedStyles(euiPinnableListGroupItemExtraActionStyles);
  var pinExtraActionLabel = useEuiI18n('euiPinnableListGroup.pinExtraActionLabel', 'Pin item');
  var pinnedExtraActionLabel = useEuiI18n('euiPinnableListGroup.pinnedExtraActionLabel', 'Unpin item');

  // Alter listItems object with extra props
  var pinnableListItems = useMemo(function () {
    return listItems.map(function (item) {
      var pinned = item.pinned,
        _item$pinnable = item.pinnable,
        pinnable = _item$pinnable === void 0 ? true : _item$pinnable,
        itemProps = _objectWithoutProperties(item, _excluded2);
      // Make some declarations of props for the nav implementation
      itemProps.className = classNames('euiPinnableListGroup__item', item.className);

      // Add the pinning action unless the item has its own extra action
      if (pinnable && !itemProps.extraAction) {
        // Different displays for pinned vs unpinned
        var sharedProps = {
          iconType: 'pinFilled',
          iconSize: 's',
          css: [styles.euiPinnableListGroup__itemExtraAction, pinned && styles.pinned]
        };
        if (pinned) {
          var title = unpinTitle ? unpinTitle(item) : pinnedExtraActionLabel;
          itemProps.extraAction = _objectSpread(_objectSpread({}, sharedProps), {}, {
            alwaysShow: true,
            'aria-label': title,
            title: title
          });
        } else {
          var _title = pinTitle ? pinTitle(item) : pinExtraActionLabel;
          itemProps.extraAction = _objectSpread(_objectSpread({}, sharedProps), {}, {
            'aria-label': _title,
            title: _title
          });
        }
        // Return the item on click
        itemProps.extraAction.onClick = function () {
          return onPinClick(item);
        };
      }
      return itemProps;
    });
  }, [listItems, pinTitle, pinExtraActionLabel, unpinTitle, pinnedExtraActionLabel, onPinClick, styles]);
  return ___EmotionJSX(EuiListGroup, _extends({
    className: classes,
    listItems: pinnableListItems
  }, rest));
};
