// @elastic/eui/es/components/tabs/tabs.styles.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS, logicalCSSWithFallback } from '../../global_styling';
import { highContrastModeStyles } from '../../global_styling/functions/high_contrast';
export var euiTabsStyles = function euiTabsStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  return {
    euiTabs: /*#__PURE__*/css("display:flex;", logicalCSS('max-width', '100%'), " ", logicalCSSWithFallback('overflow-x', 'auto'), " ", logicalCSSWithFallback('overflow-y', 'hidden'), " position:relative;flex-shrink:0;;label:euiTabs;"),
    bottomBorder: /*#__PURE__*/css(highContrastModeStyles(euiThemeContext, {
      none: "box-shadow: inset 0 -".concat(euiTheme.border.width.thin, " 0 ").concat(euiTheme.border.color, ";"),
      forced: logicalCSS('border-bottom', euiTheme.border.thin)
    }), ";;label:bottomBorder;"),
    // sizes
    s: /*#__PURE__*/css("gap:", euiTheme.size.m, ";;label:s;"),
    m: /*#__PURE__*/css("gap:", euiTheme.size.base, ";;label:m;"),
    l: /*#__PURE__*/css("gap:", euiTheme.size.l, ";;label:l;")
  };
};
