// @elastic/eui/optimize/es/components/form/form_control_layout/form_control_layout_clear_button.styles.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalSizeCSS, mathWithUnits } from '../../../global_styling';
export var EuiFormControlLayoutClearButtonStyles = function EuiFormControlLayoutClearButtonStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme,
    highContrastMode = euiThemeContext.highContrastMode;
  var backgroundColor = highContrastMode ? euiTheme.colors.darkShade : euiTheme.components.forms.clearButtonBackground;
  return {
    euiFormControlLayoutClearButton: /*#__PURE__*/css("pointer-events:all;display:flex;justify-content:center;align-items:center;border-radius:50%;border-style:solid;border-color:", backgroundColor, ";&:disabled{cursor:not-allowed;background-color:", euiTheme.colors.backgroundBaseDisabled, ";};label:euiFormControlLayoutClearButton;"),
    size: {
      s: "\n        ".concat(logicalSizeCSS(euiTheme.size.m), "\n        border-width: ").concat(mathWithUnits(euiTheme.size.m, function (x) {
        return x / 2;
      }), ";\n      "),
      m: "\n        ".concat(logicalSizeCSS(euiTheme.size.base), "\n        border-width: ").concat(mathWithUnits(euiTheme.size.base, function (x) {
        return x / 2;
      }), ";\n      ")
    },
    icon: {
      euiFormControlLayoutClearButton__icon: /*#__PURE__*/css("transform:scale(0.5);fill:", euiTheme.colors.emptyShade, ";stroke:", euiTheme.colors.emptyShade, ";;label:euiFormControlLayoutClearButton__icon;"),
      size: {
        s: /*#__PURE__*/css("stroke-width:", euiTheme.size.xs, ";;label:s;"),
        m: /*#__PURE__*/css("stroke-width:", euiTheme.size.xxs, ";;label:m;")
      }
    }
  };
};
