// @elastic/eui/optimize/es/components/form/form_control_layout/form_control_layout_delimited.styles.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS, mathWithUnits } from '../../../global_styling';
import { euiFormControlDisabledStyles, euiFormControlReadOnlyStyles, euiFormControlDefaultShadow, euiFormControlInvalidStyles, euiFormControlHoverStyles, euiFormVariables } from '../form.styles';
export var euiFormControlLayoutDelimitedStyles = function euiFormControlLayoutDelimitedStyles(euiThemeContext) {
  var highContrastMode = euiThemeContext.highContrastMode;
  var form = euiFormVariables(euiThemeContext);
  var delimitedStyles = "\n      /* Transition smoothly between disabled/readOnly background color changes */\n      ".concat(euiFormControlDefaultShadow(euiThemeContext, {
    withBorder: !highContrastMode ? true : false,
    withBackground: false
  }), "\n    ").trim();
  return {
    // Appended onto existing `euiFormControlLayout` styles
    delimited: /*#__PURE__*/css(delimitedStyles, ";label:delimited;"),
    disabled: /*#__PURE__*/css(euiFormControlDisabledStyles(euiThemeContext), ";label:disabled;"),
    readOnly: /*#__PURE__*/css(euiFormControlReadOnlyStyles(euiThemeContext), " & .euiFormControlLayoutDelimited__input{--euiFormControlStateColor:transparent;outline:none;box-shadow:none;};label:readOnly;"),
    // Appended onto existing `euiFormControlLayout__childrenWrapper` styles
    childrenWrapper: {
      delimited: /*#__PURE__*/css("display:flex;", euiFormControlDefaultShadow(euiThemeContext, {
        withBorder: !highContrastMode,
        withBackground: false
      }), " &:hover{", euiFormControlHoverStyles(euiThemeContext), " box-shadow:none;.euiFormControlLayoutDelimited__input:not(:focus){outline:none;background-color:transparent;}}>:first-child{", logicalCSS('border-top-left-radius', 'inherit'), " ", logicalCSS('border-bottom-left-radius', 'inherit'), ";}>:last-child{", logicalCSS('border-top-right-radius', 'inherit'), " ", logicalCSS('border-bottom-right-radius', 'inherit'), ";};label:delimited;"),
      invalid: /*#__PURE__*/css(euiFormControlDefaultShadow(euiThemeContext, {
        withBorder: false,
        withBackgroundColor: false
      }), "\n          :not(.euiFormControlLayoutDelimited__input, .euiFormControlLayoutDelimited__delimiter) {\n            ".concat(euiFormControlInvalidStyles(euiThemeContext), "\n          }\n\n          &:focus-within {\n            --euiFormControlStateColor: ").concat(form.borderColor, ";\n            --euiFormControlStateHoverColor: ").concat(form.borderHovered, ";\n          }\n\n          .euiFormControlLayoutDelimited__input {\n            background-color: transparent;\n          }\n        "), ";label:invalid;"),
      readOnly: /*#__PURE__*/css(";label:readOnly;")
    }
  };
};
export var euiFormControlLayoutDelimited__delimiter = function euiFormControlLayoutDelimited__delimiter(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  return /*#__PURE__*/css("display:flex;align-self:stretch;flex-grow:0;align-items:center;line-height:1;", logicalCSS('padding-horizontal', mathWithUnits([euiTheme.size.xs, euiTheme.size.xxs], function (x, y) {
    return x + y;
  })), ";;label:euiFormControlLayoutDelimited__delimiter;");
};
export var euiFormControlLayoutDelimited__input = /*#__PURE__*/css("box-shadow:none;border:none;text-align:center;", logicalCSS('height', '100%'), ";;label:euiFormControlLayoutDelimited__input;");
