// @elastic/eui/lib/services/theme/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Computed", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.Computed;
  }
});
Object.defineProperty(exports, "DEFAULT_COLOR_MODE", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.DEFAULT_COLOR_MODE;
  }
});
Object.defineProperty(exports, "buildTheme", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.buildTheme;
  }
});
Object.defineProperty(exports, "computed", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.computed;
  }
});
Object.defineProperty(exports, "getColorMode", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.getColorMode;
  }
});
Object.defineProperty(exports, "getComputed", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.getComputed;
  }
});
Object.defineProperty(exports, "getOn", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.getOn;
  }
});
Object.defineProperty(exports, "isInverseColorMode", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.isInverseColorMode;
  }
});
Object.defineProperty(exports, "mergeDeep", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.mergeDeep;
  }
});
Object.defineProperty(exports, "setOn", {
  enumerable: true,
  get: function get() {
    return _euiThemeCommon.setOn;
  }
});
var _euiThemeCommon = require("@elastic/eui-theme-common");
