// @elastic/eui/optimize/es/components/datagrid/utils/scrolling.styles.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS, mathWithUnits } from '../../../global_styling';
import { highContrastModeStyles } from '../../../global_styling/functions/high_contrast';
export var euiDataGridScrollBarStyles = function euiDataGridScrollBarStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;

  // Note that 'borders' *must* be rendered with inset box-shadow, because actual
  // `border` CSS will affect the relative position of the child scroll bar overlays
  // and cause them to be off by the width of the border
  var borderWidth = euiTheme.border.width.thin;
  var borderColor = euiTheme.border.color;
  return {
    euiDataGrid__scrollOverlay: /*#__PURE__*/css("position:absolute;inset:0;", logicalCSS('top', "-".concat(borderWidth)), "pointer-events:none;", highContrastModeStyles(euiThemeContext, {
      // Ensure the scrolling data grid body always has border edges regardless of cell position
      none: "box-shadow: inset 0 0 0 ".concat(borderWidth, " ").concat(borderColor, ";"),
      // Windows high contrast themes ignore box-shadow, so use a pseudo element workaround
      forced: "\n          &::before {\n            content: '';\n            position: absolute;\n            inset: 0;\n            ".concat(logicalCSS('border-vertical', euiTheme.border.thin), "\n          }\n          &::after {\n            content: '';\n            position: absolute;\n            inset: 0;\n            ").concat(logicalCSS('border-horizontal', euiTheme.border.thin), "\n          }\n        ")
    }), " .euiDataGrid--bordersHorizontal &{", highContrastModeStyles(euiThemeContext, {
      none: "\n            box-shadow: inset 0 -".concat(mathWithUnits(borderWidth, function (x) {
        return x * 2;
      }), " 0 -").concat(borderWidth, " ").concat(borderColor, ";\n          "),
      forced: "\n            &::after {\n              display: none;\n            }\n          "
    }), ";};label:euiDataGrid__scrollOverlay;"),
    // Ensure the horizontal scrollbar has a top border
    euiDataGrid__scrollBarOverlayBottom: /*#__PURE__*/css("position:absolute;inset-inline:0;", logicalCSS('border-top', euiTheme.border.thin), ";;label:euiDataGrid__scrollBarOverlayBottom;"),
    // Ensure the vertical scrollbar has a left border
    euiDataGrid__scrollBarOverlayRight: /*#__PURE__*/css("position:absolute;", logicalCSS('border-left', euiTheme.border.thin), ";;label:euiDataGrid__scrollBarOverlayRight;")
    // Note: Scroll bar border positions are set via JS inline style, since
    // JS has access to the exact OS scrollbar width/height and CSS doesn't
  };
};
