// @elastic/eui/optimize/es/components/selectable/selectable_templates/selectable_template_sitewide_popover.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["trigger", "search", "list", "title", "footer", "width", "isOpen", "panelRef", "closePopover"];
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { useEuiMemoizedStyles } from '../../../services';
import { EuiPopover, EuiPopoverFooter, EuiPopoverTitle } from '../../popover';
import { EuiSpacer } from '../../spacer';
import { euiSelectableTemplateSitewidePopoverStyles } from './selectable_template_sitewide.styles';
import { jsx as ___EmotionJSX } from "@emotion/react";
export var EuiSelectableTemplateSitewidePopover = function EuiSelectableTemplateSitewidePopover(_ref) {
  var trigger = _ref.trigger,
    search = _ref.search,
    list = _ref.list,
    title = _ref.title,
    footer = _ref.footer,
    width = _ref.width,
    isOpen = _ref.isOpen,
    panelRef = _ref.panelRef,
    closePopover = _ref.closePopover,
    rest = _objectWithoutProperties(_ref, _excluded);
  var styles = useEuiMemoizedStyles(euiSelectableTemplateSitewidePopoverStyles);
  return ___EmotionJSX(EuiPopover, _extends({
    panelPaddingSize: "none",
    anchorPosition: "downCenter",
    isOpen: isOpen,
    ownFocus: !!trigger,
    display: trigger ? 'inline-block' : 'block'
  }, rest, {
    panelRef: panelRef,
    button: trigger ? trigger : search,
    closePopover: closePopover,
    panelProps: {
      css: styles.euiSelectableTemplateSitewide__popover
    }
  }), ___EmotionJSX("div", {
    style: {
      width: width,
      maxWidth: '100%'
    }
  }, title || trigger ? ___EmotionJSX(EuiPopoverTitle, {
    paddingSize: "s"
  }, title, title && search && ___EmotionJSX(EuiSpacer, null), search) : undefined, list, footer && ___EmotionJSX(EuiPopoverFooter, {
    paddingSize: "s"
  }, footer)));
};
