// @elastic/eui/optimize/es/components/filter_group/filter_button.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS, mathWithUnits, euiTextShift, euiTextTruncate, highContrastModeStyles, preventForcedColors } from '../../global_styling';
import { cssSupportsHasWithNextSibling } from '../../global_styling/functions/supports';
import { euiFormVariables } from '../form/form.styles';
export var euiFilterButtonDisplay = function euiFilterButtonDisplay(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  return {
    flex: '1 1 auto',
    minInlineSize: mathWithUnits(euiTheme.size.base, function (x) {
      return x * 2.75;
    })
  };
};
var _ref4 = process.env.NODE_ENV === "production" ? {
  name: "jkp921-noGrow",
  styles: "flex-grow:0;label:noGrow;"
} : {
  name: "jkp921-noGrow",
  styles: "flex-grow:0;label:noGrow;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiFilterButtonStyles = function euiFilterButtonStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  var selectedSelector = '.euiFilterButton-isSelected';
  var withNextSelector = '& + .euiFilterButton__wrapper';
  return {
    euiFilterButton: /*#__PURE__*/css("position:relative;", logicalCSS('width', '100%'), " &:not(", selectedSelector, "){&:hover,&:active{.euiFilterButton__notification[class*='subdued']{background-color:", euiTheme.components.filterButtonBadgeBackgroundHover, ";}}};label:euiFilterButton;"),
    buttonType: {
      default: /*#__PURE__*/css("border-radius:0;&:focus-visible{z-index:1;outline-offset:-", euiTheme.border.width.thick, ";border-radius:", euiTheme.border.radius.small, ";transition:none;};label:default;"),
      toggle: /*#__PURE__*/css(euiFilterButtonDisplay(euiThemeContext), " &:focus-visible{outline-offset:", mathWithUnits(euiTheme.focus.width, function (x) {
        return x / 2;
      }), ";}", highContrastModeStyles(euiThemeContext, {
        forced: "\n            &:is(".concat(selectedSelector, ") {\n              ").concat(preventForcedColors(euiThemeContext), "\n                color: ").concat(euiTheme.colors.emptyShade, ";\n                background-color: ").concat(euiTheme.colors.fullShade, ";\n            }\n          ")
      }), ";;label:toggle;")
    },
    withNext: /*#__PURE__*/css(withNextSelector, "{", cssSupportsHasWithNextSibling("\n            &:has(+ :not(&)) {\n              ".concat(logicalCSS('padding-right', '0'), "\n            }\n          ")), ";", logicalCSS('margin-left', "-".concat(euiTheme.size.xs)), "&::before{border:none;}};label:withNext;"),
    noGrow: _ref4,
    hasNotification: /*#__PURE__*/css(logicalCSS('min-width', mathWithUnits(euiTheme.size.base, function (x) {
      return x * 6;
    })), ";;label:hasNotification;"),
    hasActiveFilters: /*#__PURE__*/css("font-weight:", euiTheme.font.weight.medium, ";;label:hasActiveFilters;")
  };
};
export var euiFilterButtonWrapperStyles = function euiFilterButtonWrapperStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  var _euiFormVariables = euiFormVariables(euiThemeContext),
    borderColor = _euiFormVariables.borderColor;
  var border = "".concat(euiTheme.border.width.thin, " solid ").concat(borderColor);

  // Pseudo elements create borders without affecting width. We also prefer them
  // over box-shadow for Windows high contrast theme compatibility
  var leftBorder = "\n    &::before {\n      content: '';\n      position: absolute;\n      inset: 0;\n      ".concat(logicalCSS('border-left', border), "\n\n      pointer-events: none;\n    }\n  ");
  // Bottom borders are needed for responsive flex-wrap behavior
  var bottomBorder = "\n    &::after {\n      content: '';\n      position: absolute;\n      inset: 0;\n      ".concat(logicalCSS('border-bottom', border), "\n      pointer-events: none;\n    }\n  ");
  return {
    wrapper: /*#__PURE__*/css(euiFilterButtonDisplay(euiThemeContext), " position:relative;display:flex;align-items:center;", leftBorder, " ", bottomBorder, " ", logicalCSS('padding-vertical', euiTheme.border.width.thin), ";;label:wrapper;"),
    hasToggle: /*#__PURE__*/css(logicalCSS('padding-horizontal', euiTheme.border.width.thin), cssSupportsHasWithNextSibling("\n          &:not([class*=\"withNext\"]):has(+ :not(&)) {\n            ".concat(logicalCSS('padding-right', '0'), "\n          }\n        ")), ";;label:hasToggle;")
  };
};
var _ref = process.env.NODE_ENV === "production" ? {
  name: "1w3amc7-disabled",
  styles: "opacity:0.5;label:disabled;"
} : {
  name: "1w3amc7-disabled",
  styles: "opacity:0.5;label:disabled;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "zxesfz-euiFilterButton__notification",
  styles: "cursor:inherit;forced-color-adjust:auto;&{transition:none;};label:euiFilterButton__notification;"
} : {
  name: "zxesfz-euiFilterButton__notification",
  styles: "cursor:inherit;forced-color-adjust:auto;&{transition:none;};label:euiFilterButton__notification;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref3 = process.env.NODE_ENV === "production" ? {
  name: "kc7mgj-euiFilterButton__content",
  styles: ".euiThemeProvider{display:inline-flex;};label:euiFilterButton__content;"
} : {
  name: "kc7mgj-euiFilterButton__content",
  styles: ".euiThemeProvider{display:inline-flex;};label:euiFilterButton__content;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiFilterButtonChildStyles = function euiFilterButtonChildStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  return {
    content: {
      euiFilterButton__content: _ref3,
      hasIcon: /*#__PURE__*/css("&>.euiIcon:last-child{", logicalCSS('margin-left', 'auto'), ";};label:hasIcon;")
    },
    text: {
      euiFilterButton__text: /*#__PURE__*/css(euiTextShift('bold', 'data-text', euiTheme), " ", euiTextTruncate(), ";;label:euiFilterButton__text;"),
      hasNotification: /*#__PURE__*/css(logicalCSS('min-width', mathWithUnits(euiTheme.size.base, function (x) {
        return x * 2;
      })), ";;label:hasNotification;")
    },
    notification: {
      euiFilterButton__notification: _ref2,
      disabled: _ref
    }
  };
};
