// @elastic/eui/optimize/es/components/markdown_editor/markdown_editor_help_button.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useState, Fragment, useContext } from 'react';
import { useEuiMemoizedStyles, useGeneratedHtmlId } from '../../services';
import { EuiButton, EuiButtonIcon } from '../button';
import { EuiTitle } from '../title';
import { EuiModal, EuiModalBody, EuiModalFooter, EuiModalHeader } from '../modal';
import { EuiI18n, useEuiI18n } from '../i18n';
import { EuiPopover } from '../popover';
import { EuiText } from '../text';
import { EuiSpacer } from '../spacer';
import { EuiToolTip } from '../tool_tip';
import { EuiHorizontalRule } from '../horizontal_rule';
import { EuiLink } from '../link';
import { EuiMarkdownContext } from './markdown_context';
// @ts-ignore a react svg
import MarkdownLogo from './icons/markdown_logo';
import { euiMarkdownEditorHelpButtonStyles } from './markdown_editor_help_button.styles';
import { jsx as ___EmotionJSX } from "@emotion/react";
var mdSyntaxHref = 'https://guides.github.com/features/mastering-markdown/';
var mdSyntaxLink = ___EmotionJSX(EuiLink, {
  href: mdSyntaxHref,
  target: "_blank"
}, ___EmotionJSX(EuiI18n, {
  token: "euiMarkdownEditorHelpButton.mdSyntaxLink",
  default: "GitHub flavored markdown"
}));
export var EuiMarkdownEditorHelpButton = function EuiMarkdownEditorHelpButton(_ref) {
  var uiPlugins = _ref.uiPlugins;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isShowingHelpModal = _useState2[0],
    setIsShowingHelpModal = _useState2[1];
  var _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    isShowingHelpPopover = _useState4[0],
    setIsShowingHelpPopover = _useState4[1];
  var _useContext = useContext(EuiMarkdownContext),
    readOnly = _useContext.readOnly;
  var styles = useEuiMemoizedStyles(euiMarkdownEditorHelpButtonStyles);
  var syntaxTitle = useEuiI18n('euiMarkdownEditorHelpButton.syntaxTitle', 'Syntax help');
  var ariaLabels = {
    showMarkdownHelp: useEuiI18n('euiMarkdownEditorHelpButton.showMarkdownHelp', 'Show markdown help')
  };
  var helpModalTitleId = useGeneratedHtmlId();
  var uiPluginsWithHelpText = uiPlugins.filter(function (_ref2) {
    var helpText = _ref2.helpText;
    return !!helpText;
  });
  var hasUiPluginsWithHelpText = uiPluginsWithHelpText.length > 0;
  if (hasUiPluginsWithHelpText) {
    return ___EmotionJSX(React.Fragment, null, ___EmotionJSX(EuiToolTip, {
      content: syntaxTitle
    }, ___EmotionJSX(EuiButtonIcon, {
      size: "s",
      css: styles.euiMarkdownEditorFooter__helpButton,
      className: "euiMarkdownEditorFooter__helpButton",
      iconType: MarkdownLogo,
      color: "text",
      "aria-label": ariaLabels.showMarkdownHelp,
      onClick: function onClick() {
        return setIsShowingHelpModal(!isShowingHelpModal);
      },
      isDisabled: readOnly
    })), isShowingHelpModal && ___EmotionJSX(EuiModal, {
      onClose: function onClose() {
        return setIsShowingHelpModal(false);
      },
      "aria-labelledby": helpModalTitleId
    }, ___EmotionJSX(EuiModalHeader, null, ___EmotionJSX(EuiTitle, null, ___EmotionJSX("h1", {
      id: helpModalTitleId
    }, syntaxTitle))), ___EmotionJSX(EuiModalBody, null, ___EmotionJSX(EuiText, null, ___EmotionJSX(EuiI18n, {
      tokens: ['euiMarkdownEditorHelpButton.syntaxModalDescriptionPrefix', 'euiMarkdownEditorHelpButton.syntaxModalDescriptionSuffix'],
      defaults: ['This editor uses', 'You can also utilize these additional syntax plugins to add rich content to your text.']
    }, function (_ref3) {
      var _ref4 = _slicedToArray(_ref3, 2),
        syntaxModalDescriptionPrefix = _ref4[0],
        syntaxModalDescriptionSuffix = _ref4[1];
      return ___EmotionJSX("p", null, syntaxModalDescriptionPrefix, " ", mdSyntaxLink, ".", ' ', syntaxModalDescriptionSuffix);
    })), ___EmotionJSX(EuiHorizontalRule, null), uiPluginsWithHelpText.map(function (_ref5) {
      var name = _ref5.name,
        helpText = _ref5.helpText;
      return ___EmotionJSX(Fragment, {
        key: name
      }, ___EmotionJSX(EuiTitle, {
        size: "xxs"
      }, ___EmotionJSX("p", null, ___EmotionJSX("strong", null, name))), ___EmotionJSX(EuiSpacer, {
        size: "s"
      }), helpText, ___EmotionJSX(EuiSpacer, {
        size: "l"
      }));
    }), ___EmotionJSX(EuiHorizontalRule, null)), ___EmotionJSX(EuiModalFooter, null, ___EmotionJSX(EuiButton, {
      onClick: function onClick() {
        return setIsShowingHelpModal(false);
      },
      fill: true
    }, ___EmotionJSX(EuiI18n, {
      token: "euiMarkdownEditorHelpButton.closeButton",
      default: "Close"
    })))));
  } else {
    return ___EmotionJSX(EuiPopover, {
      button: ___EmotionJSX(EuiButtonIcon, {
        title: syntaxTitle,
        size: "s",
        css: styles.euiMarkdownEditorFooter__helpButton,
        className: "euiMarkdownEditorFooter__helpButton",
        iconType: MarkdownLogo,
        color: "text",
        "aria-label": ariaLabels.showMarkdownHelp,
        onClick: function onClick() {
          return setIsShowingHelpPopover(!isShowingHelpPopover);
        }
      }),
      isOpen: isShowingHelpPopover,
      closePopover: function closePopover() {
        return setIsShowingHelpPopover(false);
      },
      panelPaddingSize: "s",
      anchorPosition: "upCenter",
      "aria-labelledby": helpModalTitleId
    }, ___EmotionJSX(EuiI18n, {
      tokens: ['euiMarkdownEditorHelpButton.syntaxPopoverDescription'],
      defaults: ['This editor uses']
    }, function (_ref6) {
      var _ref7 = _slicedToArray(_ref6, 1),
        syntaxPopoverDescription = _ref7[0];
      return ___EmotionJSX("p", null, syntaxPopoverDescription, " ", mdSyntaxLink, ".");
    }));
  }
};
EuiMarkdownEditorHelpButton.displayName = 'EuiMarkdownEditorHelpButton';
