// @elastic/eui/optimize/es/components/form/form_control_button/form_control_button.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS } from '../../../global_styling';
import { euiFormControlStyles, euiFormPlaceholderStyles } from '../form.styles';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "isuld2-textContent",
  styles: "flex:1;text-align:start;label:textContent;"
} : {
  name: "isuld2-textContent",
  styles: "flex:1;text-align:start;label:textContent;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiFormControlButtonStyles = function euiFormControlButtonStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  var formStyles = euiFormControlStyles(euiThemeContext);
  return {
    euiFormControlButton: /*#__PURE__*/css(formStyles.shared, " ", formStyles.uncompressed, " font-weight:", euiTheme.font.weight.regular, ";transition:none;&:hover{&::before{display:none;}}&:focus{", formStyles.focus, ";}&:disabled{", formStyles.disabled, ";};label:euiFormControlButton;"),
    isInvalid: /*#__PURE__*/css(formStyles.invalid, " &:disabled{", formStyles.invalid, ";};label:isInvalid;"),
    compressed: /*#__PURE__*/css(formStyles.compressed, ";;label:compressed;"),
    formWidth: formStyles.formWidth,
    fullWidth: /*#__PURE__*/css(formStyles.fullWidth, ";label:fullWidth;"),
    euiFormControlButton__content: /*#__PURE__*/css("justify-content:flex-start;", logicalCSS('width', '100%'), ";;label:euiFormControlButton__content;"),
    textContent: _ref,
    placeholder: /*#__PURE__*/css(euiFormPlaceholderStyles(euiThemeContext), ";;label:placeholder;")
  };
};
