// @elastic/eui/es/components/selectable/selectable_templates/selectable_template_sitewide_popover.js
var _excluded = ["trigger", "search", "list", "title", "footer", "width", "isOpen", "panelRef", "closePopover"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], t.indexOf(o) >= 0 || {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.indexOf(n) >= 0) continue; t[n] = r[n]; } return t; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import PropTypes from "prop-types";
import { useEuiMemoizedStyles } from '../../../services';
import { EuiPopover, EuiPopoverFooter, EuiPopoverTitle } from '../../popover';
import { EuiSpacer } from '../../spacer';
import { euiSelectableTemplateSitewidePopoverStyles } from './selectable_template_sitewide.styles';
import { jsx as ___EmotionJSX } from "@emotion/react";
export var EuiSelectableTemplateSitewidePopover = function EuiSelectableTemplateSitewidePopover(_ref) {
  var trigger = _ref.trigger,
    search = _ref.search,
    list = _ref.list,
    title = _ref.title,
    footer = _ref.footer,
    width = _ref.width,
    isOpen = _ref.isOpen,
    panelRef = _ref.panelRef,
    closePopover = _ref.closePopover,
    rest = _objectWithoutProperties(_ref, _excluded);
  var styles = useEuiMemoizedStyles(euiSelectableTemplateSitewidePopoverStyles);
  return ___EmotionJSX(EuiPopover, _extends({
    panelPaddingSize: "none",
    anchorPosition: "downCenter",
    isOpen: isOpen,
    ownFocus: !!trigger,
    display: trigger ? 'inline-block' : 'block'
  }, rest, {
    panelRef: panelRef,
    button: trigger ? trigger : search,
    closePopover: closePopover,
    panelProps: {
      css: styles.euiSelectableTemplateSitewide__popover
    }
  }), ___EmotionJSX("div", {
    style: {
      width: width,
      maxWidth: '100%'
    }
  }, title || trigger ? ___EmotionJSX(EuiPopoverTitle, {
    paddingSize: "s"
  }, title, title && search && ___EmotionJSX(EuiSpacer, null), search) : undefined, list, footer && ___EmotionJSX(EuiPopoverFooter, {
    paddingSize: "s"
  }, footer)));
};
EuiSelectableTemplateSitewidePopover.propTypes = {
  search: PropTypes.node.isRequired,
  list: PropTypes.node.isRequired,
  trigger: PropTypes.node,
  title: PropTypes.node,
  footer: PropTypes.node,
  width: PropTypes.any.isRequired,
  isOpen: PropTypes.bool.isRequired
};
