// @elastic/eui/optimize/lib/components/button/split_button/split_button_actions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EuiSplitButtonActionSecondary = exports.EuiSplitButtonActionPrimary = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _popover = require("../../popover");
var _tool_tip = require("../../tool_tip");
var _button = require("../button");
var _button_display = require("../button_display/_button_display");
var _button_icon = require("../button_icon");
var _split_button = require("./split_button.styles");
var _split_button_context = require("./split_button_context");
var _react2 = require("@emotion/react");
var _excluded = ["className", "isIconOnly", "tooltipProps"],
  _excluded2 = ["fill", "isDisabled", "isLoading"],
  _excluded3 = ["className", "popoverProps", "tooltipProps"],
  _excluded4 = ["fill", "isDisabled", "isLoading"];
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var EuiSplitButtonActionPrimary = exports.EuiSplitButtonActionPrimary = function EuiSplitButtonActionPrimary(_ref) {
  var className = _ref.className,
    isIconOnly = _ref.isIconOnly,
    tooltipProps = _ref.tooltipProps,
    rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _useContext = (0, _react.useContext)(_split_button_context.EuiSplitButtonContext),
    fill = _useContext.fill,
    isDisabled = _useContext.isDisabled,
    isLoading = _useContext.isLoading,
    sharedRest = (0, _objectWithoutProperties2.default)(_useContext, _excluded2);
  var _isDisabled = !!isDisabled || (0, _button_display.isButtonDisabled)(rest);
  var _isLoading = !!isLoading || !!rest.isLoading;
  var display = fill ? 'fill' : 'base';
  var classes = (0, _classnames.default)('euiSplitButtonActionPrimary', className);
  var styles = _split_button.euiSplitButtonActionStyles;
  var actionProps = _objectSpread(_objectSpread(_objectSpread({}, rest), sharedRest), {}, {
    isDisabled: _isDisabled,
    isLoading: _isLoading,
    css: [styles.euiSplitButtonActionPrimary],
    className: classes
  });
  var actionButtonProps = _objectSpread(_objectSpread({}, actionProps), {}, {
    fill: fill
  });
  var actionIconProps = _objectSpread(_objectSpread({}, actionProps), {}, {
    display: display
  });
  var button = isIconOnly ? (0, _react2.jsx)(_button_icon.EuiButtonIcon, actionIconProps) : (0, _react2.jsx)(_button.EuiButton, actionButtonProps);
  return tooltipProps ? (0, _react2.jsx)(_tool_tip.EuiToolTip, tooltipProps, button) : button;
};
var EuiSplitButtonActionSecondary = exports.EuiSplitButtonActionSecondary = function EuiSplitButtonActionSecondary(_ref2) {
  var className = _ref2.className,
    popoverProps = _ref2.popoverProps,
    tooltipProps = _ref2.tooltipProps,
    rest = (0, _objectWithoutProperties2.default)(_ref2, _excluded3);
  var _useContext2 = (0, _react.useContext)(_split_button_context.EuiSplitButtonContext),
    fill = _useContext2.fill,
    isDisabled = _useContext2.isDisabled,
    isLoading = _useContext2.isLoading,
    sharedRest = (0, _objectWithoutProperties2.default)(_useContext2, _excluded4);
  var _isDisabled = !!isDisabled || (0, _button_display.isButtonDisabled)(rest);
  var _isLoading = !!isLoading || !!rest.isLoading;
  var display = fill ? 'fill' : 'base';
  var classes = (0, _classnames.default)('euiSplitButtonActionSecondary', className);
  var styles = _split_button.euiSplitButtonActionStyles;
  var actionProps = _objectSpread(_objectSpread(_objectSpread({}, rest), sharedRest), {}, {
    display: display,
    // enforce arrowDown icon when a popover is rendered
    iconType: popoverProps != null ? 'arrowDown' : rest.iconType,
    isDisabled: _isDisabled,
    isLoading: _isLoading,
    css: [styles.euiSplitButtonActionSecondary],
    className: classes
  });
  var button = (0, _react2.jsx)(_button_icon.EuiButtonIcon, actionProps);
  var action = tooltipProps ? (0, _react2.jsx)(_tool_tip.EuiToolTip, tooltipProps, button) : button;
  return popoverProps ? (0, _react2.jsx)(_popover.EuiPopover, (0, _extends2.default)({
    anchorPosition: "downCenter"
  }, popoverProps, {
    button: action
  })) : action;
};
