// react-focus-on/dist/es5/stories/Page.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoggedIn = exports.LoggedOut = void 0;
var tslib_1 = require("tslib");
var test_1 = require("storybook/test");
var Page_1 = require("./Page");
var meta = {
    title: 'Example/Page',
    component: Page_1.Page,
    parameters: {
        // More on how to position stories at: https://storybook.js.org/docs/configure/story-layout
        layout: 'fullscreen',
    },
};
exports.default = meta;
exports.LoggedOut = {};
// More on component testing: https://storybook.js.org/docs/writing-tests/interaction-testing
exports.LoggedIn = {
    play: function (_a) { return tslib_1.__awaiter(void 0, [_a], void 0, function (_b) {
        var canvas, loginButton, logoutButton;
        var canvasElement = _b.canvasElement;
        return tslib_1.__generator(this, function (_c) {
            switch (_c.label) {
                case 0:
                    canvas = (0, test_1.within)(canvasElement);
                    loginButton = canvas.getByRole('button', { name: /Log in/i });
                    return [4 /*yield*/, (0, test_1.expect)(loginButton).toBeInTheDocument()];
                case 1:
                    _c.sent();
                    return [4 /*yield*/, test_1.userEvent.click(loginButton)];
                case 2:
                    _c.sent();
                    return [4 /*yield*/, (0, test_1.expect)(loginButton).not.toBeInTheDocument()];
                case 3:
                    _c.sent();
                    logoutButton = canvas.getByRole('button', { name: /Log out/i });
                    return [4 /*yield*/, (0, test_1.expect)(logoutButton).toBeInTheDocument()];
                case 4:
                    _c.sent();
                    return [2 /*return*/];
            }
        });
    }); },
};
