// @elastic/eui/es/components/facet/facet_button.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { euiTextTruncate, euiTextShift, logicalTextAlignCSS } from '../../global_styling';
export var euiFacetButtonStyles = function euiFacetButtonStyles(_ref) {
  var euiTheme = _ref.euiTheme;
  return {
    // Base
    euiFacetButton: /*#__PURE__*/css(logicalTextAlignCSS('left'), " &:not(:disabled){&:hover,&:focus{text-decoration:none;.euiFacetButton__text{text-decoration:underline;}}&:focus .euiFacetButton__text{text-decoration-thickness:", euiTheme.border.width.thick, ";}}&:disabled{color:", euiTheme.colors.disabledText, ";pointer-events:none;};label:euiFacetButton;")
  };
};
export var euiFacetButtonTextStyles = function euiFacetButtonTextStyles(_ref2) {
  var euiTheme = _ref2.euiTheme;
  return {
    euiFacetButton__text: /*#__PURE__*/css(euiTextShift('bold', 'data-text', euiTheme), " ", euiTextTruncate(), " flex-grow:1;;label:euiFacetButton__text;"),
    isSelected: /*#__PURE__*/css("font-weight:", euiTheme.font.weight.bold, ";;label:isSelected;"),
    unSelected: /*#__PURE__*/css(";label:unSelected;")
  };
};

// Used by both icon and quantity siblings
export var euiFacetButton__disabled = process.env.NODE_ENV === "production" ? {
  name: "1622f2l-euiFacetButton__disabled",
  styles: "opacity:0.5;label:euiFacetButton__disabled;"
} : {
  name: "1622f2l-euiFacetButton__disabled",
  styles: "opacity:0.5;label:euiFacetButton__disabled;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
