// @elastic/eui/optimize/es/components/flyout/flyout.component.js
import _extends from "@babel/runtime/helpers/extends";
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["className", "children", "as", "hideCloseButton", "flyoutMenuProps", "closeButtonProps", "closeButtonPosition", "onClose", "ownFocus", "side", "size", "paddingSize", "maxWidth", "style", "hasChildBackground", "maskProps", "type", "outsideClickCloses", "pushMinBreakpoint", "pushAnimation", "focusTrapProps", "includeFixedHeadersInFocusTrap", "includeSelectorInFocusTrap", "aria-describedby", "aria-labelledby", "id", "resizable", "minWidth", "onResize", "onAnimationEnd"],
  _excluded2 = ["titleId"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

/* eslint-disable local/i18n */

import React, { useEffect, useLayoutEffect, useRef, useMemo, useCallback, useState, forwardRef } from 'react';
import classnames from 'classnames';
import { keys, EuiWindowEvent, useCombinedRefs, useEuiMemoizedStyles, useGeneratedHtmlId, useEuiThemeCSSVariables, focusTrapPubSub } from '../../services';
import { useCurrentSession, useIsInManagedFlyout, useFlyoutLayoutMode, useFlyoutId, useFlyoutWidth, useIsFlyoutActive, useFlyoutManager, useHasPushPadding } from './manager';
import { LAYOUT_MODE_STACKED } from './manager/const';
import { EuiFocusTrap } from '../focus_trap';
import { EuiI18n } from '../i18n';
import { useResizeObserver } from '../observer/resize_observer';
import { EuiScreenReaderOnly } from '../accessibility';
import { EuiFlyoutCloseButton } from './_flyout_close_button';
import { euiFlyoutStyles, composeFlyoutInlineStyles } from './flyout.styles';
import { usePropsWithComponentDefaults } from '../provider/component_defaults';
import { DEFAULT_PADDING_SIZE, DEFAULT_PUSH_MIN_BREAKPOINT, DEFAULT_SIDE, DEFAULT_SIZE, DEFAULT_TYPE, isEuiFlyoutSizeNamed } from './const';
import { useIsPushed } from './hooks';
import { EuiFlyoutMenu } from './flyout_menu';
import { EuiFlyoutOverlay } from './_flyout_overlay';
import { EuiFlyoutResizeButton } from './_flyout_resize_button';
import { useEuiFlyoutResizable } from './use_flyout_resizable';
import { useEuiFlyoutZIndex } from './use_flyout_z_index';
import { EuiFlyoutParentProvider } from './flyout_parent_context';
import { useCurrentFlyoutZIndexRef } from './manager/selectors';
import { jsx as ___EmotionJSX } from "@emotion/react";
var defaultElement = 'div';
export var EuiFlyoutComponent = /*#__PURE__*/forwardRef(function (props, ref) {
  var _usePropsWithComponen = usePropsWithComponentDefaults('EuiFlyout', props),
    className = _usePropsWithComponen.className,
    children = _usePropsWithComponen.children,
    as = _usePropsWithComponen.as,
    _usePropsWithComponen2 = _usePropsWithComponen.hideCloseButton,
    hideCloseButton = _usePropsWithComponen2 === void 0 ? false : _usePropsWithComponen2,
    _flyoutMenuProps = _usePropsWithComponen.flyoutMenuProps,
    closeButtonProps = _usePropsWithComponen.closeButtonProps,
    _usePropsWithComponen3 = _usePropsWithComponen.closeButtonPosition,
    closeButtonPosition = _usePropsWithComponen3 === void 0 ? 'inside' : _usePropsWithComponen3,
    onClose = _usePropsWithComponen.onClose,
    _usePropsWithComponen4 = _usePropsWithComponen.ownFocus,
    ownFocus = _usePropsWithComponen4 === void 0 ? true : _usePropsWithComponen4,
    _usePropsWithComponen5 = _usePropsWithComponen.side,
    side = _usePropsWithComponen5 === void 0 ? DEFAULT_SIDE : _usePropsWithComponen5,
    _usePropsWithComponen6 = _usePropsWithComponen.size,
    _size = _usePropsWithComponen6 === void 0 ? DEFAULT_SIZE : _usePropsWithComponen6,
    _usePropsWithComponen7 = _usePropsWithComponen.paddingSize,
    paddingSize = _usePropsWithComponen7 === void 0 ? DEFAULT_PADDING_SIZE : _usePropsWithComponen7,
    _usePropsWithComponen8 = _usePropsWithComponen.maxWidth,
    maxWidth = _usePropsWithComponen8 === void 0 ? false : _usePropsWithComponen8,
    style = _usePropsWithComponen.style,
    _usePropsWithComponen9 = _usePropsWithComponen.hasChildBackground,
    hasChildBackground = _usePropsWithComponen9 === void 0 ? false : _usePropsWithComponen9,
    maskProps = _usePropsWithComponen.maskProps,
    _usePropsWithComponen10 = _usePropsWithComponen.type,
    type = _usePropsWithComponen10 === void 0 ? DEFAULT_TYPE : _usePropsWithComponen10,
    outsideClickCloses = _usePropsWithComponen.outsideClickCloses,
    _usePropsWithComponen11 = _usePropsWithComponen.pushMinBreakpoint,
    pushMinBreakpoint = _usePropsWithComponen11 === void 0 ? DEFAULT_PUSH_MIN_BREAKPOINT : _usePropsWithComponen11,
    _usePropsWithComponen12 = _usePropsWithComponen.pushAnimation,
    pushAnimation = _usePropsWithComponen12 === void 0 ? false : _usePropsWithComponen12,
    _focusTrapProps = _usePropsWithComponen.focusTrapProps,
    _usePropsWithComponen13 = _usePropsWithComponen.includeFixedHeadersInFocusTrap,
    includeFixedHeadersInFocusTrap = _usePropsWithComponen13 === void 0 ? true : _usePropsWithComponen13,
    includeSelectorInFocusTrap = _usePropsWithComponen.includeSelectorInFocusTrap,
    _ariaDescribedBy = _usePropsWithComponen['aria-describedby'],
    _ariaLabelledBy = _usePropsWithComponen['aria-labelledby'],
    id = _usePropsWithComponen.id,
    _usePropsWithComponen14 = _usePropsWithComponen.resizable,
    resizable = _usePropsWithComponen14 === void 0 ? false : _usePropsWithComponen14,
    minWidth = _usePropsWithComponen.minWidth,
    onResize = _usePropsWithComponen.onResize,
    onAnimationEnd = _usePropsWithComponen.onAnimationEnd,
    rest = _objectWithoutProperties(_usePropsWithComponen, _excluded);
  var _useEuiThemeCSSVariab = useEuiThemeCSSVariables(),
    setGlobalCSSVariables = _useEuiThemeCSSVariab.setGlobalCSSVariables;
  var Element = as || defaultElement;
  var maskRef = useRef(null);

  // Ref for the main flyout element to pass to context
  var internalParentFlyoutRef = useRef(null);
  var isPushed = useIsPushed({
    type: type,
    pushMinBreakpoint: pushMinBreakpoint
  });
  var currentSession = useCurrentSession();
  var isInManagedContext = useIsInManagedFlyout();
  var flyoutId = useFlyoutId(id);
  var layoutMode = useFlyoutLayoutMode();
  var isActiveManagedFlyout = useIsFlyoutActive(flyoutId);
  var flyoutManager = useFlyoutManager();
  var currentZIndexRef = useCurrentFlyoutZIndexRef();

  // Use a ref to access the latest flyoutManager without triggering effect re-runs
  var flyoutManagerRef = useRef(flyoutManager);
  useEffect(function () {
    flyoutManagerRef.current = flyoutManager;
  }, [flyoutManager]);
  useEffect(function () {
    // Keep track of unmanaged flyouts to properly calculate z-index
    // values for all flyouts
    if (!isInManagedContext) {
      var _flyoutManagerRef$cur;
      (_flyoutManagerRef$cur = flyoutManagerRef.current) === null || _flyoutManagerRef$cur === void 0 || _flyoutManagerRef$cur.addUnmanagedFlyout(flyoutId);
      return function () {
        var _flyoutManagerRef$cur2;
        return (_flyoutManagerRef$cur2 = flyoutManagerRef.current) === null || _flyoutManagerRef$cur2 === void 0 ? void 0 : _flyoutManagerRef$cur2.closeUnmanagedFlyout(flyoutId);
      };
    }
  }, [isInManagedContext, flyoutId]);
  var _useEuiFlyoutResizabl = useEuiFlyoutResizable({
      enabled: resizable,
      minWidth: minWidth,
      maxWidth: typeof maxWidth === 'number' ? maxWidth : 0,
      onResize: onResize,
      side: side,
      size: _size
    }),
    onMouseDownResizableButton = _useEuiFlyoutResizabl.onMouseDown,
    onKeyDownResizableButton = _useEuiFlyoutResizabl.onKeyDown,
    size = _useEuiFlyoutResizabl.size,
    setFlyoutRef = _useEuiFlyoutResizabl.setFlyoutRef;

  /**
   * Setting up the refs on the actual flyout element in order to
   * accommodate for the `isPushed` state by adding padding to the body equal to the width of the element
   */
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    resizeRef = _useState2[0],
    setResizeRef = _useState2[1];
  var setRef = useCombinedRefs([setResizeRef, ref, internalParentFlyoutRef, setFlyoutRef]);
  var _useResizeObserver = useResizeObserver(isPushed ? resizeRef : null, 'width'),
    width = _useResizeObserver.width;

  // Memoize flyout identification and relationships to prevent race conditions
  var flyoutIdentity = useMemo(function () {
    if (!flyoutId || !currentSession) {
      return {
        isMainFlyout: false,
        siblingFlyoutId: null,
        hasValidSession: false,
        sessionForWidth: null
      };
    }
    var siblingFlyoutId = currentSession.mainFlyoutId === flyoutId ? currentSession.childFlyoutId : currentSession.mainFlyoutId;
    return {
      isMainFlyout: currentSession.mainFlyoutId === flyoutId,
      siblingFlyoutId: siblingFlyoutId,
      hasValidSession: true,
      sessionForWidth: currentSession
    };
  }, [flyoutId, currentSession]);

  // Destructure for easier use
  var siblingFlyoutId = flyoutIdentity.siblingFlyoutId,
    isMainFlyout = flyoutIdentity.isMainFlyout;
  var _siblingFlyoutWidth = useFlyoutWidth(siblingFlyoutId);
  var siblingFlyoutWidth = layoutMode === LAYOUT_MODE_STACKED ? 0 : _siblingFlyoutWidth;

  /**
   * Effect for adding push padding to body. Using useLayoutEffect to ensure
   * padding changes happen synchronously before child components render -
   * this is needed to prevent RemoveScrollBar from measuring the body in an
   * inconsistent state during flyout transitions.
   */
  useLayoutEffect(function () {
    if (!isPushed) {
      return; // Only push-type flyouts manage body padding
    }
    var shouldApplyPadding = !isInManagedContext || isActiveManagedFlyout;
    var paddingSide = side === 'left' ? 'paddingInlineStart' : 'paddingInlineEnd';
    var cssVarName = "--euiPushFlyoutOffset".concat(side === 'left' ? 'InlineStart' : 'InlineEnd');
    var managerSide = side === 'left' ? 'left' : 'right';
    var paddingWidth = layoutMode === LAYOUT_MODE_STACKED && isMainFlyout && _siblingFlyoutWidth ? _siblingFlyoutWidth : width;
    if (shouldApplyPadding) {
      document.body.style[paddingSide] = "".concat(paddingWidth, "px");
      setGlobalCSSVariables(_defineProperty({}, cssVarName, "".concat(paddingWidth, "px")));
      // Update manager state if in managed context
      if (isInManagedContext && flyoutManagerRef.current) {
        flyoutManagerRef.current.setPushPadding(managerSide, paddingWidth);
      }
    } else {
      // Explicitly remove padding when this push flyout becomes inactive
      document.body.style[paddingSide] = '';
      setGlobalCSSVariables(_defineProperty({}, cssVarName, null));
      // Clear manager state if in managed context
      if (isInManagedContext && flyoutManagerRef.current) {
        flyoutManagerRef.current.setPushPadding(managerSide, 0);
      }
    }

    // Cleanup on unmount
    return function () {
      document.body.style[paddingSide] = '';
      setGlobalCSSVariables(_defineProperty({}, cssVarName, null));
      // Clear manager state on unmount if in managed context
      if (isInManagedContext && flyoutManagerRef.current) {
        flyoutManagerRef.current.setPushPadding(managerSide, 0);
      }
    };
  }, [isPushed, isInManagedContext, isActiveManagedFlyout, setGlobalCSSVariables, side, width, layoutMode, isMainFlyout, _siblingFlyoutWidth]);

  /**
   * This class doesn't actually do anything by EUI, but is nice to add for consumers (JIC)
   */
  useEffect(function () {
    document.body.classList.add('euiBody--hasFlyout');
    return function () {
      // Remove the hasFlyout class when the flyout is unmounted
      document.body.classList.remove('euiBody--hasFlyout');
    };
  }, []);
  var hasChildFlyout = (currentSession === null || currentSession === void 0 ? void 0 : currentSession.childFlyoutId) != null;
  var isChildFlyout = isInManagedContext && hasChildFlyout && (currentSession === null || currentSession === void 0 ? void 0 : currentSession.childFlyoutId) === id;
  var shouldCloseOnEscape = useMemo(function () {
    // Regular flyout - always close on ESC
    if (!isInManagedContext) {
      return true;
    }

    // Managed flyout with no child - close on ESC
    if (!hasChildFlyout) {
      return true;
    }

    // Child flyout - close on ESC
    if (isChildFlyout) {
      return true;
    }

    // Main flyout with child flyout - don't close on ESC
    return false;
  }, [isInManagedContext, hasChildFlyout, isChildFlyout]);

  /**
   * ESC key closes flyout based on flyout hierarchy rules
   */
  var onKeyDown = useCallback(function (event) {
    if (!isPushed && event.key === keys.ESCAPE && shouldCloseOnEscape) {
      event.preventDefault();
      onClose(event);
    }
  }, [onClose, isPushed, shouldCloseOnEscape]);
  var managedFlyoutIndex = currentZIndexRef.current;
  if (isInManagedContext && currentSession) {
    managedFlyoutIndex = currentSession.zIndex;
  }
  var _useEuiFlyoutZIndex = useEuiFlyoutZIndex({
      maskProps: maskProps,
      isPushed: isPushed,
      managedFlyoutIndex: managedFlyoutIndex,
      isChildFlyout: isChildFlyout
    }),
    flyoutZIndex = _useEuiFlyoutZIndex.flyoutZIndex,
    maskZIndex = _useEuiFlyoutZIndex.maskZIndex;

  /**
   * Set inline styles
   */
  var inlineStyles = useMemo(function () {
    var composedStyles = composeFlyoutInlineStyles(size, layoutMode, siblingFlyoutId, siblingFlyoutWidth || null, maxWidth, flyoutZIndex);
    return _objectSpread(_objectSpread({}, style), composedStyles);
  }, [style, size, layoutMode, siblingFlyoutId, siblingFlyoutWidth, maxWidth, flyoutZIndex]);
  var styles = useEuiMemoizedStyles(euiFlyoutStyles);
  var cssStyles = [styles.euiFlyout, styles.paddingSizes[paddingSize], isEuiFlyoutSizeNamed(size) && styles[size], maxWidth === false && styles.noMaxWidth, isPushed ? styles.push.push : styles.overlay.overlay, isPushed ? styles.push[side] : styles.overlay[side], isPushed && !pushAnimation && styles.push.noAnimation, styles[side]];
  var classes = classnames('euiFlyout', isChildFlyout && hasChildBackground && 'euiFlyout--hasChildBackground', className);
  var flyoutToggle = useRef(document.activeElement);
  var _useState3 = useState([]),
    _useState4 = _slicedToArray(_useState3, 2),
    focusTrapShards = _useState4[0],
    setFocusTrapShards = _useState4[1];
  var focusTrapSelectors = useMemo(function () {
    var selectors = [];
    if (includeSelectorInFocusTrap) {
      selectors = Array.isArray(includeSelectorInFocusTrap) ? includeSelectorInFocusTrap : [includeSelectorInFocusTrap];
    }
    if (includeFixedHeadersInFocusTrap) {
      selectors.push('.euiHeader[data-fixed-header]');
    }
    return selectors;
  }, [includeSelectorInFocusTrap, includeFixedHeadersInFocusTrap]);

  /**
   * Finds the shards to include in the focus trap by querying by `focusTrapSelectors`.
   *
   * @param shouldAutoFocus Whether to auto-focus the flyout wrapper when the focus trap is activated.
   * This is necessary because when a flyout is toggled from within a shard, the focus trap's `autoFocus`
   * feature doesn't work. This logic manually focuses the flyout as a workaround.
   */
  var findShards = useCallback(function () {
    var shouldAutoFocus = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    if (focusTrapSelectors.length > 0) {
      var shardsEls = focusTrapSelectors.flatMap(function (selector) {
        return Array.from(document.querySelectorAll(selector));
      });
      setFocusTrapShards(Array.from(shardsEls));
      if (shouldAutoFocus) {
        shardsEls.forEach(function (shard) {
          if (shard.contains(flyoutToggle.current)) {
            resizeRef === null || resizeRef === void 0 || resizeRef.focus();
          }
        });
      }
    } else {
      // Clear existing shards if necessary, e.g. switching to `false`
      setFocusTrapShards(function (shards) {
        return shards.length ? [] : shards;
      });
    }
  }, [focusTrapSelectors, resizeRef]);
  useEffect(function () {
    // Auto-focus should only happen on initial flyout mount (or when the dependencies change)
    // because it snaps focus to the flyout wrapper, which steals it from subsequently focused elements.
    findShards(true);
    var unsubscribe = focusTrapPubSub.subscribe(function () {
      return findShards();
    });
    return unsubscribe;
  }, [findShards]);
  var focusTrapProps = useMemo(function () {
    return _objectSpread(_objectSpread({}, _focusTrapProps), {}, {
      shards: [].concat(_toConsumableArray(focusTrapShards), _toConsumableArray((_focusTrapProps === null || _focusTrapProps === void 0 ? void 0 : _focusTrapProps.shards) || []))
    });
  }, [_focusTrapProps, focusTrapShards]);

  /*
   * Provide meaningful screen reader instructions/details
   */
  var hasOverlayMask = ownFocus && !isPushed;
  var descriptionId = useGeneratedHtmlId();
  var ariaDescribedBy = classnames(descriptionId, _ariaDescribedBy);
  var screenReaderDescription = useMemo(function () {
    return ___EmotionJSX(EuiScreenReaderOnly, null, ___EmotionJSX("p", {
      id: descriptionId
    }, hasOverlayMask ? ___EmotionJSX(EuiI18n, {
      token: "euiFlyout.screenReaderModalDialog",
      default: "You are in a modal dialog. Press Escape or tap/click outside the dialog on the shadowed overlay to close."
    }) : ___EmotionJSX(EuiI18n, {
      token: "euiFlyout.screenReaderNonModalDialog",
      default: "You are in a non-modal dialog. To close the dialog, press Escape."
    }), ' ', focusTrapShards.length > 0 && ___EmotionJSX(EuiI18n, {
      token: "euiFlyout.screenReaderFocusTrapShards",
      default: "You can still continue tabbing through other global page landmarks."
    })));
  }, [hasOverlayMask, descriptionId, focusTrapShards.length]);

  /*
   * If the flyout menu is to be rendered, ensure the flyout has aria-labelledby referencing the menu's titleId
   */
  var generatedMenuId = useGeneratedHtmlId();
  var _ref = _flyoutMenuProps || {},
    _titleId = _ref.titleId,
    flyoutMenuProps = _objectWithoutProperties(_ref, _excluded2);
  var hasMenu = !!_flyoutMenuProps;
  var flyoutMenuId = useMemo(function () {
    if (!hasMenu) return undefined;
    return _titleId || generatedMenuId;
  }, [hasMenu, _titleId, generatedMenuId]);

  // If the flyout level is LEVEL_MAIN, the title should be hidden by default
  var flyoutMenuHideTitle = useMemo(function () {
    if (!hasMenu) return undefined;
    if ((_flyoutMenuProps === null || _flyoutMenuProps === void 0 ? void 0 : _flyoutMenuProps.hideTitle) !== undefined) {
      return _flyoutMenuProps.hideTitle;
    }
    return (currentSession === null || currentSession === void 0 ? void 0 : currentSession.mainFlyoutId) === flyoutId;
  }, [hasMenu, _flyoutMenuProps, currentSession, flyoutId]);
  var ariaLabelledBy = useMemo(function () {
    if (flyoutMenuId) {
      return classnames(flyoutMenuId, _ariaLabelledBy);
    }
    return _ariaLabelledBy;
  }, [flyoutMenuId, _ariaLabelledBy]);

  /*
   * Trap focus even when `ownFocus={false}`, otherwise closing
   * the flyout won't return focus to the originating button.
   *
   * Set `clickOutsideDisables={true}` when `ownFocus={false}`
   * to allow non-keyboard users the ability to interact with
   * elements outside the flyout.
   *
   * Set `onClickOutside={onClose}` when `ownFocus` and `type` are the defaults,
   * or if `outsideClickCloses={true}` to close on clicks that target
   * (both mousedown and mouseup) the overlay mask.
   */
  var onClickOutside = useCallback(function (event) {
    // Do not close the flyout for any external click
    if (outsideClickCloses === false) return undefined;
    if (hasOverlayMask) {
      // The overlay mask is present, so only clicks on the mask should close the flyout, regardless of outsideClickCloses
      if (event.target === maskRef.current) return onClose(event);
    } else {
      // No overlay mask is present, so any outside clicks should close the flyout
      if (outsideClickCloses === true) return onClose(event);
    }
    // Otherwise if ownFocus is false and outsideClickCloses is undefined, outside clicks should not close the flyout
    return undefined;
  }, [onClose, hasOverlayMask, outsideClickCloses]);
  var maskCombinedRefs = useCombinedRefs([maskProps === null || maskProps === void 0 ? void 0 : maskProps.maskRef, maskRef]);

  /**
   * For overlay flyouts in managed contexts, coordinate scroll locking with push flyout state.
   */
  var hasPushPaddingInManager = useHasPushPadding();
  var shouldDeferScrollLock = !isPushed && isInManagedContext && hasPushPaddingInManager;
  var shouldUseScrollLock = hasOverlayMask && !shouldDeferScrollLock;
  return ___EmotionJSX(EuiFlyoutOverlay, {
    hasOverlayMask: hasOverlayMask,
    isPushed: isPushed,
    maskZIndex: maskZIndex,
    maskProps: _objectSpread(_objectSpread({}, maskProps), {}, {
      maskRef: maskCombinedRefs
    })
  }, ___EmotionJSX(EuiWindowEvent, {
    event: "keydown",
    handler: onKeyDown
  }), ___EmotionJSX(EuiFocusTrap, _extends({
    disabled: isPushed,
    scrollLock: shouldUseScrollLock,
    clickOutsideDisables: !ownFocus,
    onClickOutside: onClickOutside
  }, focusTrapProps), ___EmotionJSX(Element, _extends({
    className: classes,
    css: cssStyles,
    style: inlineStyles,
    ref: setRef,
    id: id
  }, rest, {
    role: !isPushed ? 'dialog' : rest.role,
    "aria-modal": !isPushed || undefined,
    tabIndex: !isPushed ? 0 : rest.tabIndex,
    "aria-describedby": !isPushed ? ariaDescribedBy : _ariaDescribedBy,
    "aria-labelledby": ariaLabelledBy,
    "data-autofocus": !isPushed || undefined,
    onAnimationEnd: onAnimationEnd
  }), !isPushed && screenReaderDescription, !_flyoutMenuProps && !hideCloseButton && ___EmotionJSX(EuiFlyoutCloseButton, _extends({}, closeButtonProps, {
    onClose: onClose,
    closeButtonPosition: closeButtonPosition,
    side: side
  })), _flyoutMenuProps && ___EmotionJSX(EuiFlyoutMenu, _extends({}, flyoutMenuProps, {
    hideTitle: flyoutMenuHideTitle,
    titleId: flyoutMenuId
  })), resizable && ___EmotionJSX(EuiFlyoutResizeButton, {
    type: type,
    side: side,
    ownFocus: ownFocus,
    isPushed: isPushed,
    onMouseDown: onMouseDownResizableButton,
    onTouchStart: onMouseDownResizableButton,
    onKeyDown: onKeyDownResizableButton
  }), ___EmotionJSX(EuiFlyoutParentProvider, null, children))));
}
// React.forwardRef interferes with the inferred element type
// Casting to ensure correct element prop type checking for `as`
// e.g., `href` is not on a `div`
);
// Recast to allow `displayName`
EuiFlyoutComponent.displayName = 'EuiFlyoutComponent';
