// @elastic/eui-theme-common/lib/cjs/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buildTheme = exports.DEFAULT_COLOR_MODE = exports.Computed = void 0;
exports.computed = computed;
exports.formatMultipleBoxShadow = formatMultipleBoxShadow;
exports.setOn = exports.mergeDeep = exports.isInverseColorMode = exports.getTokenName = exports.getOn = exports.getComputed = exports.getColorMode = void 0;
var _chromaJs = _interopRequireDefault(require("chroma-js"));
var _global_styling = require("./global_styling");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toConsumableArray(r) { return _arrayWithoutHoles(r) || _iterableToArray(r) || _unsupportedIterableToArray(r) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _iterableToArray(r) { if ("undefined" != typeof Symbol && null != r[Symbol.iterator] || null != r["@@iterator"]) return Array.from(r); }
function _arrayWithoutHoles(r) { if (Array.isArray(r)) return _arrayLikeToArray(r); }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); } /*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
var DEFAULT_COLOR_MODE = exports.DEFAULT_COLOR_MODE = _global_styling.COLOR_MODES_STANDARD.light;

/**
 * Returns whether the parameter is an object
 * @param {any} obj - Anything
 */
var isObject = function isObject(obj) {
  return obj && _typeof(obj) === 'object';
};

/**
 * Returns whether the provided color mode is `inverse`
 * @param {string} colorMode - `light`, `dark`, or `inverse`
 */
var isInverseColorMode = exports.isInverseColorMode = function isInverseColorMode(colorMode) {
  return colorMode === _global_styling.COLOR_MODES_INVERSE;
};

/**
 * Returns the color mode configured in the current EuiThemeProvider.
 * Returns the parent color mode if none is explicity set.
 * @param {string} colorMode - `light`, `dark`, or `inverse`
 * @param {string} parentColorMode - `LIGHT` or `DARK`; used as the fallback
 * @param {boolean} isForced
 */
var getColorMode = exports.getColorMode = function getColorMode(colorMode, parentColorMode, isForced) {
  if (isForced || colorMode == null) {
    return parentColorMode || DEFAULT_COLOR_MODE;
  }
  var mode = colorMode.toUpperCase();
  if (isInverseColorMode(mode)) {
    return parentColorMode === _global_styling.COLOR_MODES_STANDARD.dark || parentColorMode === undefined ? _global_styling.COLOR_MODES_STANDARD.light : _global_styling.COLOR_MODES_STANDARD.dark;
  } else {
    return mode;
  }
};

/**
 * Returns a value at a given path on an object.
 * If `colorMode` is provided, will scope the value to the appropriate color mode key (LIGHT\DARK)
 * @param {object} model - Object
 * @param {string} _path - Dot-notated string to a path on the object
 * @param {string} colorMode - `light` or `dark`
 */
var getOn = exports.getOn = function getOn(model, _path, colorMode) {
  var path = _path.split('.');
  var node = model;
  while (path.length) {
    var segment = path.shift();
    if (node.hasOwnProperty(segment) === false) {
      if (colorMode && node.hasOwnProperty(colorMode) === true && node[colorMode].hasOwnProperty(segment) === true) {
        if (node[colorMode][segment] instanceof Computed) {
          node = node[colorMode][segment].getValue(null, {}, node, colorMode);
        } else {
          node = node[colorMode][segment];
        }
      } else {
        return undefined;
      }
    } else {
      if (node[segment] instanceof Computed) {
        node = node[segment].getValue(null, {}, node, colorMode);
      } else {
        node = node[segment];
      }
    }
  }
  return node;
};

/**
 * Sets a value at a given path on an object.
 * @param {object} model - Object
 * @param {string} _path - Dot-notated string to a path on the object
 * @param {any} string -  The value to set
 */
var setOn = exports.setOn = function setOn(model, _path, value) {
  var path = _path.split('.');
  var propertyName = path.pop();
  var node = model;
  while (path.length) {
    var segment = path.shift();
    if (node.hasOwnProperty(segment) === false) {
      node[segment] = {};
    }
    node = node[segment];
  }
  node[propertyName] = value;
  return true;
};

/**
 * Creates a class to store the `computer` method and its eventual parameters.
 * Allows for on-demand computation with up-to-date parameters via `getValue` method.
 * @constructor
 * @param {function} computer - Function to be computed
 * @param {string | array} dependencies - Dependencies passed to the `computer` as parameters
 */
var Computed = exports.Computed = /*#__PURE__*/function () {
  function Computed(computer) {
    var dependencies = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
    _classCallCheck(this, Computed);
    this.computer = computer;
    this.dependencies = dependencies;
  }

  /**
   * Executes the `computer` method with the current state of the theme
   * by taking into account previously computed values and modifications.
   * @param {Proxy | object} base - Computed or uncomputed theme
   * @param {Proxy | object} modifications - Theme value overrides
   * @param {object} working - Partially computed theme
   * @param {string} colorMode - `light` or `dark`
   */
  return _createClass(Computed, [{
    key: "getValue",
    value: function getValue(base) {
      var modifications = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
      var working = arguments.length > 2 ? arguments[2] : undefined;
      var colorMode = arguments.length > 3 ? arguments[3] : undefined;
      if (!this.dependencies.length) {
        return this.computer(working);
      }
      if (!Array.isArray(this.dependencies)) {
        var _ref, _getOn;
        return this.computer((_ref = (_getOn = getOn(working, this.dependencies)) !== null && _getOn !== void 0 ? _getOn : getOn(modifications, this.dependencies, colorMode)) !== null && _ref !== void 0 ? _ref : base ? getOn(base, this.dependencies, colorMode) : working);
      }
      return this.computer(this.dependencies.map(function (dependency) {
        var _ref2, _getOn2;
        return (_ref2 = (_getOn2 = getOn(working, dependency)) !== null && _getOn2 !== void 0 ? _getOn2 : getOn(modifications, dependency, colorMode)) !== null && _ref2 !== void 0 ? _ref2 : base ? getOn(base, dependency, colorMode) : working;
      }));
    }
  }]);
}();
/**
 * Returns a Class (`Computed`) that stores the arbitrary computer method
 * and references to its optional dependecies.
 * @param {function} computer - Arbitrary method to be called at compute time.
 * @param {string | array} dependencies - Values that will be provided to `computer` at compute time.
 */
function computed(comp, dep) {
  return new Computed(comp, dep);
}

/**
 * Type guard to check for a Computed object based on object shape
 * without relying on the Computed class directly
 */
var isComputedLike = function isComputedLike(key) {
  if (_typeof(key) !== 'object' || Array.isArray(key)) return false;
  return key.hasOwnProperty('dependencies') && key.hasOwnProperty('computer');
};

/**
 * Takes an uncomputed theme, and computes and returns all values taking
 * into consideration value overrides and configured color mode.
 * Overrides take precedence, and only values in the current color mode
 * are computed and returned.
 * @param {Proxy} base - Object to transform into Proxy
 * @param {Proxy | object} over - Unique identifier or name
 * @param {string} colorMode - `light` or `dark`
 */
var getComputed = exports.getComputed = function getComputed(base, over, colorMode, highContrastMode) {
  var output = {
    themeName: base.key
  };
  var _hcmOverridesKey = "".concat(_global_styling.EUI_THEME_OVERRIDES_KEY, ".").concat(_global_styling.EUI_THEME_HIGH_CONTRAST_MODE_KEY);
  var _hcmBaseOverrides = highContrastMode ? getOn(base, _hcmOverridesKey) : undefined;
  var _hcmOverOverrides = highContrastMode ? getOn(over, _hcmOverridesKey) : undefined;
  function loop(base, over) {
    var checkExisting = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
    var path = arguments.length > 3 ? arguments[3] : undefined;
    Object.keys(base).forEach(function (key) {
      var newPath = path ? "".concat(path, ".").concat(key) : "".concat(key);

      // remove the internal overrides key from the computed theme object
      // the override values are only used internally in getComputed()
      if (key === _global_styling.EUI_THEME_OVERRIDES_KEY) {
        return;
      }

      // @ts-expect-error `key` is not necessarily a colorMode key
      if ([].concat(_toConsumableArray(Object.values(_global_styling.COLOR_MODES_STANDARD)), [colorMode]).includes(key)) {
        if (key !== colorMode) {
          return;
        } else {
          var colorModeSegment = new RegExp("(\\.".concat(colorMode, "\\b)|(\\b").concat(colorMode, "\\.)"));
          newPath = newPath.replace(colorModeSegment, '');
        }
      }
      var existing = checkExisting && getOn(output, newPath);
      if (!existing || isObject(existing)) {
        // NOTE: the class type check for Computed is not true for themes created externally;
        // we additionally check on the object shape to confirm a Computed value
        var baseValue = base[key] instanceof Computed || isComputedLike(base[key]) ? base[key].getValue(base.root, over.root, output, colorMode) : base[key];
        var overValue = over[key] instanceof Computed || isComputedLike(over[key]) ? over[key].getValue(base.root, over.root, output, colorMode) : over[key];
        var hcmBaseValue = _hcmBaseOverrides ? _hcmBaseOverrides[key] instanceof Computed || isComputedLike(_hcmBaseOverrides[key]) ? _hcmBaseOverrides[key].getValue(base.root, _hcmBaseOverrides.root, output, colorMode) : _hcmBaseOverrides[key] : undefined;
        var hcmOverValue = _hcmOverOverrides ? _hcmOverOverrides[key] instanceof Computed || isComputedLike(_hcmOverOverrides[key]) ? _hcmOverOverrides[key].getValue(base.root, _hcmOverOverrides.root, output, colorMode) : _hcmOverOverrides[key] : undefined;
        var hcmCombinedOverValue = hcmOverValue !== null && hcmOverValue !== void 0 ? hcmOverValue : hcmBaseValue;

        // combine internal overrides with manual overrides
        var combinedOverValue = isObject(overValue) && isObject(hcmCombinedOverValue) ? mergeDeep(overValue, hcmCombinedOverValue) : // optional overrides e.g. on provider level should still override theme level
        overValue !== null && overValue !== void 0 ? overValue : hcmCombinedOverValue;
        if (isObject(baseValue) && !Array.isArray(baseValue)) {
          loop(baseValue, combinedOverValue !== null && combinedOverValue !== void 0 ? combinedOverValue : {}, checkExisting, newPath);
        } else {
          setOn(output, newPath, combinedOverValue !== null && combinedOverValue !== void 0 ? combinedOverValue : baseValue);
        }
      }
    });
  }
  // Compute standard theme values and apply overrides
  loop(base, over);
  // Compute and apply extension values only
  loop(over, {}, true);
  return output;
};

/**
 * Builds a Proxy with a custom `handler` designed to self-reference values
 * and prevent arbitrary value overrides.
 * @param {object} model - Object to transform into Proxy
 * @param {string} key - Unique identifier or name
 */
var buildTheme = exports.buildTheme = function buildTheme(model, key) {
  var handler = {
    getPrototypeOf: function getPrototypeOf(target) {
      return Reflect.getPrototypeOf(target.model);
    },
    setPrototypeOf: function setPrototypeOf(target, prototype) {
      return Reflect.setPrototypeOf(target.model, prototype);
    },
    isExtensible: function isExtensible(target) {
      return Reflect.isExtensible(target);
    },
    preventExtensions: function preventExtensions(target) {
      return Reflect.preventExtensions(target.model);
    },
    getOwnPropertyDescriptor: function getOwnPropertyDescriptor(target, key) {
      return Reflect.getOwnPropertyDescriptor(target.model, key);
    },
    defineProperty: function defineProperty(target, property, attributes) {
      return Reflect.defineProperty(target.model, property, attributes);
    },
    has: function has(target, property) {
      return Reflect.has(target.model, property);
    },
    get: function get(_target, property) {
      if (property === 'key') {
        return _target[property];
      }

      // prevent Safari from locking up when the proxy is used in dev tools
      // as it doesn't support getPrototypeOf
      if (property === '__proto__') return {};
      var target = property === 'root' ? _target : _target.model || _target;
      // @ts-ignore `string` index signature
      var value = target[property];
      if (isObject(value) && !Array.isArray(value)) {
        return new Proxy({
          model: value,
          root: _target.root,
          key: "_".concat(_target.key)
        }, handler);
      } else {
        return value;
      }
    },
    set: function set(target) {
      return target;
    },
    deleteProperty: function deleteProperty(target) {
      return target;
    },
    ownKeys: function ownKeys(target) {
      return Reflect.ownKeys(target.model);
    },
    apply: function apply(target) {
      return target;
    },
    construct: function construct(target) {
      return target;
    }
  };
  var themeProxy = new Proxy({
    model: model,
    root: model,
    key: key
  }, handler);
  return themeProxy;
};

/**
 * Deeply merges two objects, using `source` values whenever possible.
 * @param {object} _target - Object with fallback values
 * @param {object} source - Object with desired values
 */
var mergeDeep = exports.mergeDeep = function mergeDeep(_target) {
  var source = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  var target = _objectSpread({}, _target);
  if (!isObject(target) || !isObject(source)) {
    return source;
  }
  Object.keys(source).forEach(function (key) {
    var targetValue = target[key];
    var sourceValue = source[key];
    if (isObject(targetValue) && isObject(sourceValue)) {
      target[key] = mergeDeep(_objectSpread({}, targetValue), _objectSpread({}, sourceValue));
    } else {
      target[key] = sourceValue;
    }
  });
  return target;
};

/**
 * Returns token name string based on passed dynamic variants
 * and additional prefix/suffix
 */
var getTokenName = exports.getTokenName = function getTokenName(prefix, variant, suffix) {
  var getCapitalized = function getCapitalized(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
  };
  var _suffix = suffix ? getCapitalized(suffix) : '';
  if (!variant) {
    return "".concat(prefix).concat(_suffix);
  }
  var colorName = variant.charAt(0).toUpperCase() + variant.slice(1);
  return "".concat(prefix).concat(getCapitalized(colorName)).concat(_suffix);
};

/**
 * Format an array of shadow "objects" into a string for CSS.
 * The "up" direction is built by making the y offset from layers
 * two and any subsequent layers, negative.
 *
 * It will add `inset` to the first layer when color mode is "DARK"
 * and `spread` is 1, to ensure the dark mode-only floating border
 * is rendered inside the container to match the position of the
 * regular panel border (implemented with a pseudo element).
 *
 * @todo add a color space param to replace the hard-coded `hsl`
 *
 * @param layers
 * @param options
 * @param options.up - Modifies some values in order to get the "up" direction
 * @returns - A value for the CSS `box-shadow` property
 */
function formatMultipleBoxShadow(layers, options) {
  if (layers.length === 0) {
    return 'none';
  }
  var _ref3 = options !== null && options !== void 0 ? options : {},
    _ref3$direction = _ref3.direction,
    direction = _ref3$direction === void 0 ? 'down' : _ref3$direction,
    _ref3$color = _ref3.color,
    color = _ref3$color === void 0 ? '#000' : _ref3$color;
  /* prettier-ignore */
  var shadowLayers = layers.map(function (layer, i) {
    var y = direction === 'up' && i > 0 ? -layer.y : layer.y;
    var shadowColor = (0, _chromaJs["default"])(color).alpha(layer.opacity).css('hsl');
    return "".concat(layer.x, "px ").concat(y, "px ").concat(layer.blur, "px ").concat(layer.spread, "px ").concat(shadowColor);
  });
  return shadowLayers.join(', ');
}
