// @elastic/eui/es/components/flyout/flyout_menu.js
var _excluded = ["className", "title", "titleId", "hideTitle", "hideCloseButton", "historyItems", "showBackButton", "backButtonProps", "customActions"];
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], t.indexOf(o) >= 0 || {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.indexOf(n) >= 0) continue; t[n] = r[n]; } return t; }
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
import PropTypes from "prop-types";
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import classNames from 'classnames';
import React, { useContext, useState } from 'react';
import { useEuiMemoizedStyles } from '../../services';
import { EuiButtonEmpty, EuiButtonIcon } from '../button';
import { EuiFlexGroup, EuiFlexItem } from '../flex';
import { EuiListGroup, EuiListGroupItem } from '../list_group';
import { EuiPopover } from '../popover';
import { EuiTitle } from '../title';
import { EuiFlyoutCloseButton } from './_flyout_close_button';
import { euiFlyoutMenuStyles } from './flyout_menu.styles';
import { EuiFlyoutMenuContext } from './flyout_menu_context';
import { EuiI18n, useEuiI18n } from '../i18n';

/**
 * History item for the flyout menu history popover
 */

/**
 * Custom action item for the flyout menu component
 */

/**
 * Props for EuiFlyoutMenu
 */
import { jsx as ___EmotionJSX } from "@emotion/react";
var BackButton = function BackButton(props) {
  return ___EmotionJSX(EuiButtonEmpty, _extends({
    size: "xs",
    color: "text",
    iconType: "editorUndo"
  }, props), ___EmotionJSX(EuiI18n, {
    token: "euiFlyoutMenu.back",
    default: "Back"
  }));
};
var HistoryPopover = function HistoryPopover(_ref) {
  var items = _ref.items;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isPopoverOpen = _useState2[0],
    setIsPopoverOpen = _useState2[1];
  var handlePopoverButtonClick = function handlePopoverButtonClick() {
    setIsPopoverOpen(!isPopoverOpen);
  };
  return ___EmotionJSX(EuiPopover, {
    button: ___EmotionJSX(EuiButtonIcon, {
      iconType: "arrowDown",
      color: "text",
      "aria-label": useEuiI18n('euiFlyoutMenu.history', 'History')
    }),
    isOpen: isPopoverOpen,
    onClick: handlePopoverButtonClick,
    closePopover: function closePopover() {
      return setIsPopoverOpen(false);
    },
    panelPaddingSize: "xs",
    anchorPosition: "downLeft"
  }, ___EmotionJSX(EuiListGroup, {
    gutterSize: "none"
  }, items.map(function (item, index) {
    return ___EmotionJSX(EuiListGroupItem, {
      key: "history-item-".concat(index),
      label: item.title,
      size: "s",
      onClick: function onClick() {
        item.onClick();
        setIsPopoverOpen(false);
      }
    }, item.title);
  })));
};

/**
 * The component for the top menu bar inside a flyout. Since this is a private
 * component, rendering is controlled using the `flyoutMenuProps` prop on
 * `EuiFlyout`. In managed session flyouts, the Flyout Manager controls a back
 * button and history popover for navigating to different flyout sessions
 * within the managed context.
 *
 * @private
 */
export var EuiFlyoutMenu = function EuiFlyoutMenu(_ref2) {
  var className = _ref2.className,
    title = _ref2.title,
    titleId = _ref2.titleId,
    hideTitle = _ref2.hideTitle,
    hideCloseButton = _ref2.hideCloseButton,
    _ref2$historyItems = _ref2.historyItems,
    historyItems = _ref2$historyItems === void 0 ? [] : _ref2$historyItems,
    showBackButton = _ref2.showBackButton,
    backButtonProps = _ref2.backButtonProps,
    customActions = _ref2.customActions,
    rest = _objectWithoutProperties(_ref2, _excluded);
  var _useContext = useContext(EuiFlyoutMenuContext),
    onClose = _useContext.onClose;
  var styles = useEuiMemoizedStyles(euiFlyoutMenuStyles);
  var classes = classNames('euiFlyoutMenu', className);
  var titleNode;
  if (title) {
    titleNode = ___EmotionJSX(EuiTitle, {
      size: "xxs",
      id: titleId
    }, ___EmotionJSX("h3", {
      css: hideTitle && styles.euiFlyoutMenu__hiddenTitle
    }, title));
  }
  var handleClose = function handleClose(event) {
    onClose === null || onClose === void 0 || onClose(event);
  };
  var closeButton = ___EmotionJSX(EuiFlyoutCloseButton, {
    onClose: handleClose,
    side: "right",
    closeButtonPosition: "inside"
  });
  return ___EmotionJSX("div", _extends({
    className: classes,
    css: styles.euiFlyoutMenu__container
  }, rest), ___EmotionJSX(EuiFlexGroup, {
    alignItems: "center",
    justifyContent: "spaceBetween",
    gutterSize: "none",
    responsive: false
  }, showBackButton && ___EmotionJSX(EuiFlexItem, {
    grow: false
  }, ___EmotionJSX(BackButton, backButtonProps)), historyItems.length > 0 && ___EmotionJSX(EuiFlexItem, {
    grow: false
  }, ___EmotionJSX(HistoryPopover, {
    items: historyItems
  })), titleNode && ___EmotionJSX(EuiFlexItem, {
    grow: false
  }, titleNode), ___EmotionJSX(EuiFlexItem, {
    grow: true
  }), customActions && customActions.map(function (action, actionIndex) {
    return ___EmotionJSX(EuiFlexItem, {
      grow: false,
      key: "action-index-flex-item-".concat(actionIndex),
      css: styles.euiFlyoutMenu__actions
    }, ___EmotionJSX(EuiButtonIcon, {
      key: "action-index-icon-".concat(actionIndex),
      "aria-label": action['aria-label'],
      iconType: action.iconType,
      onClick: action.onClick,
      color: "text",
      size: "s"
    }));
  }), !hideCloseButton && ___EmotionJSX(EuiFlexItem, {
    grow: false,
    css: styles.euiFlyoutMenu__spacer
  })), !hideCloseButton && closeButton);
};
EuiFlyoutMenu.propTypes = {
  className: PropTypes.string,
  "aria-label": PropTypes.string,
  "data-test-subj": PropTypes.string,
  css: PropTypes.any,
  /**
       * An id to use for the title element. Useful for setting aria-labelledby on the flyout.
       * Example:
       * ```jsx
       * <EuiFlyout
       *   aria-labelledby="myMenuTitleId"
       *   flyoutMenuProps={{ title: 'Menu title', titleId: 'myMenuTitleId' }
       * >
       *  ...
       * </EuiFlyout>
       * ```
       */
  titleId: PropTypes.string,
  /**
       * Title for the menu component. In a managed flyout context, the title is used to indicate the flyout session for history navigation.
       */
  title: PropTypes.node,
  /**
       * Hides the title in the `EuiFlyoutMenu`. This is useful when the title is already shown in an `EuiFlyoutHeader`.
       * @default true for main flyout in a managed flyout session; false otherwise
       */
  hideTitle: PropTypes.bool,
  /**
       * Hides the close button in the menu component
       * @default false
       */
  hideCloseButton: PropTypes.bool,
  /**
       * Shows a back button in the menu component
       * @default false
       */
  showBackButton: PropTypes.bool,
  /**
       * Props to pass to the back button, such as `onClick` handler
       */
  backButtonProps: PropTypes.any,
  /**
       * List of history items for the history popover
       */
  historyItems: PropTypes.arrayOf(PropTypes.shape({
    /**
       * Title for the history item
       */
    title: PropTypes.string.isRequired,
    /**
       * onClick handler for the history item
       */
    onClick: PropTypes.func.isRequired
  }).isRequired),
  /**
       * List of custom action items for the menu component
       */
  customActions: PropTypes.arrayOf(PropTypes.shape({
    /**
       * Icon type for the action button
       */
    iconType: PropTypes.string.isRequired,
    /**
       * onClick handler for the action button
       */
    onClick: PropTypes.func.isRequired,
    /**
       * Aria label for the action button
       */
    "aria-label": PropTypes.string.isRequired
  }).isRequired)
};
