// @elastic/eui/es/components/flyout/index.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

export { EuiFlyout } from './flyout';
export { EuiFlyoutBody } from './flyout_body';
export { EuiFlyoutFooter } from './flyout_footer';
export { EuiFlyoutHeader } from './flyout_header';
export { euiFlyoutSlideInRight, euiFlyoutSlideInLeft } from './flyout.styles';
export { EuiFlyoutResizable } from './flyout_resizable';
export { EuiFlyoutMenu } from './flyout_menu';

// Hooks for using Manager-based flyouts
export { useIsInManagedFlyout, useHasActiveSession } from './manager';
export { useIsInsideParentFlyout } from './flyout_parent_context';

// Flyout manager store (for cross-root state synchronization)
export { getFlyoutManagerStore } from './manager';
