// @elastic/eui/es/components/flyout/manager/const.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

/**
 * Allowed values for `session` prop to control the way the session is managed for a flyout.
 */
export var SESSION_START = 'start';
export var SESSION_INHERIT = 'inherit';
export var SESSION_NEVER = 'never';
var PREFIX = 'data-managed-flyout';

/**
 * Data attribute applied to a managed flyout element to help identify it as a managed flyout.
 */
export var PROPERTY_FLYOUT = "".concat(PREFIX);

/**
 * Data attribute indicating whether the flyout is the `main` or `child` flyout.
 */
export var PROPERTY_LEVEL = "".concat(PREFIX, "-level");

/**
 * Data attribute representing how multiple flyouts are laid out
 * (`side-by-side` or `stacked`).
 */
export var PROPERTY_LAYOUT_MODE = "".concat(PREFIX, "-layout-mode");

/** Stacked layout mode where child flyouts overlay on top of the main flyout. */
export var LAYOUT_MODE_STACKED = 'stacked';
/** Side-by-side layout mode where child flyouts render adjacent to the main flyout. */
export var LAYOUT_MODE_SIDE_BY_SIDE = 'side-by-side';

/** The primary (parent) flyout in a session. */
export var LEVEL_MAIN = 'main';
/** The secondary (child) flyout spawned by the main flyout. */
export var LEVEL_CHILD = 'child';

/** Flyout is mounting and playing its opening animation. */
export var STAGE_OPENING = 'opening';
/** Flyout is fully visible and interactive. */
export var STAGE_ACTIVE = 'active';
/** Flyout is unmounted or not currently visible/interactable. */
export var STAGE_INACTIVE = 'inactive';
/** Main flyout is transitioning behind an active session flyout. */
export var STAGE_BACKGROUNDING = 'backgrounding';
/** Main flyout is backgrounded behind an active session flyout. */
export var STAGE_BACKGROUNDED = 'backgrounded';
/** Flyout is returning to the foreground from a backgrounded state. */
export var STAGE_RETURNING = 'returning';
/** Flyout is playing its closing animation. */
export var STAGE_CLOSING = 'closing';
