// @elastic/eui/es/components/flyout/manager/flyout_managed.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["id", "onClose", "onActive", "level", "size", "css", "flyoutMenuProps"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], t.indexOf(o) >= 0 || {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.indexOf(n) >= 0) continue; t[n] = r[n]; } return t; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useEffect, useLayoutEffect, useMemo, useRef, useState, forwardRef } from 'react';
import PropTypes from "prop-types";
import { flushSync } from 'react-dom';
import { useCombinedRefs, useEuiMemoizedStyles } from '../../../services';
import { useEuiI18n } from '../../i18n';
import { useResizeObserver } from '../../observer/resize_observer';
import { EuiFlyoutComponent } from '../flyout.component';
import { EuiFlyoutMenuContext } from '../flyout_menu_context';
import { useFlyoutActivityStage } from './activity_stage';
import { LEVEL_CHILD, LEVEL_MAIN, PROPERTY_FLYOUT, PROPERTY_LAYOUT_MODE, PROPERTY_LEVEL } from './const';
import { EuiFlyoutIsManagedProvider } from './context';
import { euiManagedFlyoutStyles } from './flyout_managed.styles';
import { useFlyoutManager as _useFlyoutManager, useCurrentSession, useFlyoutId, useFlyoutLayoutMode, useIsFlyoutActive, useParentFlyoutSize } from './hooks';
import { useCurrentMainFlyout, useIsFlyoutRegistered } from './selectors';
import { createValidationErrorMessage, isNamedSize, validateManagedFlyoutSize, validateSizeCombination } from './validation';

/**
 * Props for `EuiManagedFlyout`, the internal persistent flyout used by
 * the manager. Extends base flyout props and requires a `level` to
 * distinguish `main` vs `child` behavior.
 */
import { jsx as ___EmotionJSX } from "@emotion/react";
var useFlyoutManager = function useFlyoutManager() {
  var context = _useFlyoutManager();
  if (!context) {
    throw new Error('EuiManagedFlyout must be used within an EuiFlyoutManager');
  }
  return context;
};

/**
 * Persistent managed flyout rendered inside the provider. Handles:
 * - registration/unregistration with the manager
 * - size validation and parent/child size compatibility
 * - width tracking for responsive layouts
 * - lifecycle stage transitions and data attributes for styling
 */
export var EuiManagedFlyout = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var id = _ref.id,
    onCloseProp = _ref.onClose,
    onActiveProp = _ref.onActive,
    level = _ref.level,
    sizeProp = _ref.size,
    customCss = _ref.css,
    _flyoutMenuProps = _ref.flyoutMenuProps,
    props = _objectWithoutProperties(_ref, _excluded);
  var flyoutId = useFlyoutId(id);
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    flyoutRef = _useState2[0],
    setFlyoutRef = _useState2[1];
  var refs = useMemo(function () {
    return [setFlyoutRef, ref];
  }, [ref]);
  var combinedRef = useCombinedRefs(refs);
  var _useFlyoutManager2 = useFlyoutManager(),
    addFlyout = _useFlyoutManager2.addFlyout,
    closeFlyout = _useFlyoutManager2.closeFlyout,
    setFlyoutWidth = _useFlyoutManager2.setFlyoutWidth,
    goBack = _useFlyoutManager2.goBack,
    _historyItems = _useFlyoutManager2.historyItems;
  var parentSize = useParentFlyoutSize(flyoutId);
  var parentFlyout = useCurrentMainFlyout();
  var layoutMode = useFlyoutLayoutMode();
  var styles = useEuiMemoizedStyles(euiManagedFlyoutStyles);

  // Set default size based on level: main defaults to 'm', child defaults to 's'
  var size = sizeProp !== null && sizeProp !== void 0 ? sizeProp : level === LEVEL_CHILD ? 's' : 'm';

  // Validate size
  var sizeTypeError = validateManagedFlyoutSize(size, flyoutId, level);
  if (sizeTypeError) {
    throw new Error(createValidationErrorMessage(sizeTypeError));
  }

  // For child flyouts, validate parent-child combinations
  if (level === LEVEL_CHILD && parentSize && isNamedSize(size) && isNamedSize(parentSize)) {
    var combinationError = validateSizeCombination(parentSize, size);
    if (combinationError) {
      combinationError.flyoutId = flyoutId;
      combinationError.parentFlyoutId = parentFlyout === null || parentFlyout === void 0 ? void 0 : parentFlyout.flyoutId;
      combinationError.level = level;
      throw new Error(createValidationErrorMessage(combinationError));
    }
  }
  var defaultTitle = useEuiI18n('euiFlyoutManaged.defaultTitle', 'Unknown Flyout');

  // Set title from flyoutMenuProps or aria-label
  // TODO: allow aria-labelledby references to be used
  var title = (_flyoutMenuProps === null || _flyoutMenuProps === void 0 ? void 0 : _flyoutMenuProps.title) || props['aria-label'];
  if (process.env.NODE_ENV === 'development' && level === LEVEL_MAIN && !title) {
    console.warn("Managed flyout \"".concat(flyoutId, "\" requires a title, which can be provided through 'flyoutMenuProps.title' or 'aria-label'. Using default title: \"").concat(defaultTitle, "\""));
    title = defaultTitle;
  }
  var isActive = useIsFlyoutActive(flyoutId);
  var currentSession = useCurrentSession();
  var flyoutExistsInManager = useIsFlyoutRegistered(flyoutId);

  // Stabilize the onClose callback
  var onCloseCallbackRef = useRef();
  onCloseCallbackRef.current = function (e) {
    if (onCloseProp) {
      var event = e || new MouseEvent('click');
      onCloseProp(event);
    }
  };

  // Stabilize the onActive callback
  var onActiveCallbackRef = useRef();
  onActiveCallbackRef.current = onActiveProp;

  // Track if flyout was ever registered to avoid false positives on initial mount
  var wasRegisteredRef = useRef(false);

  // Track flyoutExistsInManager in a ref to avoid dependency loop
  // The cleanup function needs the current value but shouldn't cause re-runs
  var flyoutExistsInManagerRef = useRef(flyoutExistsInManager);

  // Update ref when flyoutExistsInManager changes
  useEffect(function () {
    flyoutExistsInManagerRef.current = flyoutExistsInManager;
  }, [flyoutExistsInManager]);

  // Register with flyout manager context when open, remove when closed
  // Using useLayoutEffect to run synchronously before DOM updates
  useLayoutEffect(function () {
    addFlyout(flyoutId, title, level, size);
    return function () {
      // Only call closeFlyout if it wasn't already called via onClose
      // This prevents duplicate removal when using Escape/X button
      if (flyoutExistsInManagerRef.current) {
        closeFlyout(flyoutId);
      }

      // Reset navigation tracking when explicitly closed via isOpen=false
      wasRegisteredRef.current = false;
    };
  }, [flyoutId, title, level, size, addFlyout, closeFlyout]);

  // Detect when flyout has been removed from manager state (e.g., via Back button)
  // and trigger onClose callback to notify the parent component
  useEffect(function () {
    if (flyoutExistsInManager) {
      wasRegisteredRef.current = true;
    }

    // If flyout was previously registered, is marked as open, but no longer exists in manager state,
    // it was removed via navigation (Back button) - trigger close callback
    if (wasRegisteredRef.current && !flyoutExistsInManager) {
      var _onCloseCallbackRef$c;
      (_onCloseCallbackRef$c = onCloseCallbackRef.current) === null || _onCloseCallbackRef$c === void 0 || _onCloseCallbackRef$c.call(onCloseCallbackRef, new MouseEvent('navigation'));
      wasRegisteredRef.current = false; // Reset to avoid repeated calls
    }
  }, [flyoutExistsInManager, flyoutId]);

  // Monitor current session changes and fire onActive callback when this flyout becomes active
  useEffect(function () {
    if (!onActiveCallbackRef.current || !currentSession) {
      return;
    }

    // Make sure callback is only fired for the flyout that changed
    var mainChanged = level === LEVEL_MAIN && currentSession.mainFlyoutId === flyoutId;
    var childChanged = level === LEVEL_CHILD && currentSession.childFlyoutId === flyoutId;
    if (mainChanged || childChanged) {
      onActiveCallbackRef.current();
    }
  }, [currentSession, flyoutId, level]);

  // Track width changes for flyouts
  var _useResizeObserver = useResizeObserver(isActive ? flyoutRef : null, 'width'),
    width = _useResizeObserver.width;

  // Pass the stabilized onClose callback to the flyout menu context
  var onClose = function onClose(e) {
    // CRITICAL: Update manager state FIRST before allowing React to unmount
    // This prevents race conditions during portal → inline DOM transitions
    // and ensures cascade close logic runs before DOM cleanup begins
    // Using flushSync to force synchronous state update completion
    flushSync(function () {
      closeFlyout(flyoutId);
    });

    // trigger parent callback, unmounts the component
    if (onCloseCallbackRef.current) {
      var event = e || new MouseEvent('click');
      onCloseCallbackRef.current(event);
    }
  };

  // Update width in manager state when it changes
  useEffect(function () {
    if (isActive && width) {
      setFlyoutWidth(flyoutId, width);
    }
  }, [flyoutId, level, isActive, width, setFlyoutWidth]);
  var _useFlyoutActivitySta = useFlyoutActivityStage({
      flyoutId: flyoutId,
      level: level
    }),
    activityStage = _useFlyoutActivitySta.activityStage,
    onAnimationEnd = _useFlyoutActivitySta.onAnimationEnd;

  // Note: history controls are only relevant for main flyouts
  var historyItems = useMemo(function () {
    var result = level === LEVEL_MAIN ? _historyItems : undefined;
    return result;
  }, [level, _historyItems]);
  var backButtonProps = useMemo(function () {
    return level === LEVEL_MAIN ? {
      onClick: goBack
    } : undefined;
  }, [level, goBack]);
  var showBackButton = historyItems ? historyItems.length > 0 : false;
  var flyoutMenuProps = _objectSpread(_objectSpread({}, _flyoutMenuProps), {}, {
    historyItems: historyItems,
    showBackButton: showBackButton,
    backButtonProps: backButtonProps,
    title: title
  });
  return ___EmotionJSX(EuiFlyoutIsManagedProvider, {
    isManaged: true
  }, ___EmotionJSX(EuiFlyoutMenuContext.Provider, {
    value: {
      onClose: onClose
    }
  }, ___EmotionJSX(EuiFlyoutComponent, _extends({
    id: flyoutId,
    ref: combinedRef,
    css: [styles.managedFlyout, customCss, styles.stage(activityStage, props.side, level), ";label:EuiManagedFlyout;"]
  }, _objectSpread(_objectSpread({}, props), {}, _defineProperty(_defineProperty(_defineProperty({
    onClose: onClose,
    size: size,
    flyoutMenuProps: flyoutMenuProps,
    onAnimationEnd: onAnimationEnd
  }, PROPERTY_FLYOUT, true), PROPERTY_LAYOUT_MODE, layoutMode), PROPERTY_LEVEL, level))))));
});
EuiManagedFlyout.propTypes = {
  level: PropTypes.oneOfType([PropTypes.any.isRequired, PropTypes.any.isRequired]).isRequired,
  flyoutMenuProps: PropTypes.shape({
    className: PropTypes.string,
    "aria-label": PropTypes.string,
    "data-test-subj": PropTypes.string,
    css: PropTypes.any,
    /**
         * An id to use for the title element. Useful for setting aria-labelledby on the flyout.
         * Example:
         * ```jsx
         * <EuiFlyout
         *   aria-labelledby="myMenuTitleId"
         *   flyoutMenuProps={{ title: 'Menu title', titleId: 'myMenuTitleId' }
         * >
         *  ...
         * </EuiFlyout>
         * ```
         */
    titleId: PropTypes.string,
    /**
         * Title for the menu component. In a managed flyout context, the title is used to indicate the flyout session for history navigation.
         */
    title: PropTypes.node,
    /**
         * Hides the title in the `EuiFlyoutMenu`. This is useful when the title is already shown in an `EuiFlyoutHeader`.
         * @default true for main flyout in a managed flyout session; false otherwise
         */
    hideTitle: PropTypes.bool,
    /**
         * Hides the close button in the menu component
         * @default false
         */
    hideCloseButton: PropTypes.bool,
    /**
         * Props to pass to the back button, such as `onClick` handler
         */
    backButtonProps: PropTypes.any,
    /**
         * List of custom action items for the menu component
         */
    customActions: PropTypes.arrayOf(PropTypes.shape({
      /**
         * Icon type for the action button
         */
      iconType: PropTypes.string.isRequired,
      /**
         * onClick handler for the action button
         */
      onClick: PropTypes.func.isRequired,
      /**
         * Aria label for the action button
         */
      "aria-label": PropTypes.string.isRequired
    }).isRequired)
  }),
  onActive: PropTypes.func
};
EuiManagedFlyout.displayName = 'EuiManagedFlyout';
